{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2dbf888a-f549-40f4-b53f-26cb535f2040",
   "metadata": {},
   "source": [
    "# See if the optimizer can function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "2f1366e7-5864-4829-8ff5-1c688acf7548",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:test\n"
     ]
    }
   ],
   "source": [
    "from nomelt.thermo_estimation.optimizer import MutationSubsetOptimizer, OptimizerArgs\n",
    "from nomelt.thermo_estimation.estimators import ESMFoldDGEstimator, ESMFoldDGArgs\n",
    "from nomelt.thermo_estimation.rosetta import RosettaMinimizationParameters\n",
    "import torch\n",
    "import optuna\n",
    "import os\n",
    "\n",
    "import logging\n",
    "logger = logging.getLogger('root')\n",
    "logger.setLevel(logging.DEBUG)\n",
    "logging.info('test')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0fef0b19-e043-4a90-8f8e-0d71538c5a57",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "torch.cuda.is_available()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "08474d90-8f57-4bb0-b3b6-a6542462b60f",
   "metadata": {},
   "outputs": [],
   "source": [
    "rosetta_params = RosettaMinimizationParameters(update_pdb=True)\n",
    "estimator_args = ESMFoldDGArgs(rosetta_params=rosetta_params)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ecd42d1-80b4-490a-85c7-5990b4441213",
   "metadata": {},
   "source": [
    "### Use the enh1 and concesus homeodomain as a ground truth test case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c3cf65ce-00a5-4d29-804a-80e23bccb0d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "_1enh = \"RPRTAFSSEQLARLKREFNENRYLTERRRQQLSSELGLNEAQIKIWFQNKRAKI\"\n",
    "_concensus = \"RRKRTTFTKEQLEELEELFEKNRYPSAEEREELAKKLGLTERQVKVWFQNRRAKEKK\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "747f5f91-2138-4768-a829-b428f664f303",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 2 proteins.\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.85s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.707861 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/wt.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 54\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.210554 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/wt.pdb with minimized positions\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/conc.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 57\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/conc.pdb with minimized positions\n"
     ]
    }
   ],
   "source": [
    "estimator = ESMFoldDGEstimator(sequences=[_1enh, _concensus], ids=['wt', 'conc'], args=estimator_args)\n",
    "ground_results = estimator.run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b138cc7d-d11d-4223-8920-7825c121d0d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'wt': -130.62494306511954, 'conc': -140.3217549930466}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ground_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0cdca63-3933-478f-afbf-44c23e4c0cbc",
   "metadata": {},
   "source": [
    "#### run the optimizer to see if it can find a better one than the concesus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8941be75-0fe8-4238-b4f8-3ef8c42d2cea",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:nomelt.thermo_estimation.optimizer:Aligned sequences to determined mutation set:\n",
      "INFO:nomelt.thermo_estimation.optimizer:\n",
      "R-PRTAFSSEQLARLKRE-FNE-NRYLT-ERRRQQLSSELGLNEAQIKIWFQNKRA--KI\n",
      "| .||.|..|||..|. | | | |||.. | .|..|...|||.|.|.|.||||.||  |.\n",
      "RRKRTTFTKEQLEELE-ELF-EKNRYPSAE-EREELAKKLGLTERQVKVWFQNRRAKEKK\n",
      "  Score=149\n",
      "\n",
      "INFO:nomelt.thermo_estimation.optimizer:Aligned sequences with mutations marked:\n",
      "INFO:nomelt.thermo_estimation.optimizer:\n",
      "R |  - |  P | RT |  A | F |  S |  S | EQL |  A |  R | L |  K |  R | E |  - | F \n",
      "= 0    1    | == 2    | = 3    4    | === 5    6    | = 7    8    | = 9    | = \n",
      "R |  R |  K | RT |  T | F |  T |  K | EQL |  E |  E | L |  E |  - | E |  L | F \n",
      "\n",
      "|  N | E |  - | NRY |  L |  T |  - | E |  R |  R | R |  Q |  Q | L |  S |  S | \n",
      "10   | = 11   | === 12   13   14   | = 15   16   | = 17   18   | = 19   20   21\n",
      "|  - | E |  K | NRY |  P |  S |  A | E |  - |  E | R |  E |  E | L |  A |  K | \n",
      "\n",
      " E | LGL |  N | E |  A | Q |  I | K |  I | WFQN |  K | RA |  - |  - | K |  I\n",
      "   | === 22   | = 23   | = 24   | = 25   | ==== 26   | == 27   28   | = 29  \n",
      " K | LGL |  T | E |  R | Q |  V | K |  V | WFQN |  R | RA |  K |  E | K |  K\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "args = OptimizerArgs(\n",
    "    n_trials=100,\n",
    "    gapopen=-2,\n",
    "    penalize_end_gaps=True,\n",
    "    matrix='BLOSUM62',\n",
    "    sampler = optuna.samplers.NSGAIISampler(\n",
    "        population_size=10\n",
    "    )\n",
    "    \n",
    "\n",
    ")\n",
    "optimizer = MutationSubsetOptimizer(\n",
    "    _1enh, _concensus, \n",
    "    estimator=ESMFoldDGEstimator,\n",
    "    args=args,\n",
    "    estimator_args=estimator_args\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "57647e2e-6b30-4eb5-820c-758a38de0b8e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:24:07,107] A new study created in memory with name: mutation_optimizer_run\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "83671ee4582440b3b7993a969fb2b3a0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/100 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_22', 'mut_27', 'mut_30', 'mut_42', 'mut_46', 'mut_48', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLARLKREFNEKNRYLSERRQQLSSELGLTEAQVKVWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n",
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:06<00:00,  6.27s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-505218220 seed_offset=0 real_seed=-505218220\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-505218220 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.740321 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5468023129485296103.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 57\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.193752 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5468023129485296103.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_2', 'mut_7', 'mut_13', 'mut_18', 'mut_20', 'mut_28', 'mut_30', 'mut_31', 'mut_37', 'mut_38', 'mut_44', 'mut_53', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RKRTAFTSEQLAELKRELFENRYLTAEERQQLSKKLGLNERQIKIWFQNRRAEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:24:42,502] Trial 0 finished with value: -118.65175757290251 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': False, 'mut_16': False, 'mut_18': False, 'mut_20': False, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': False, 'mut_30': True, 'mut_31': False, 'mut_33': False, 'mut_34': False, 'mut_36': False, 'mut_37': False, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': True, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 0 with value: -118.65175757290251.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.47s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1279968392 seed_offset=0 real_seed=-1279968392\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1279968392 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.632903 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7595516037513144332.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 55\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188131 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7595516037513144332.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_2', 'mut_7', 'mut_12', 'mut_16', 'mut_18', 'mut_20', 'mut_26', 'mut_27', 'mut_28', 'mut_31', 'mut_34', 'mut_42', 'mut_44', 'mut_46', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RKRTAFTSEQLERLKELFENRYPSAERERQELSSELGLTERQVKIWFQNKRAEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:24:47,338] Trial 1 finished with value: -127.01452559072571 and parameters: {'mut_1': False, 'mut_2': True, 'mut_5': False, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': False, 'mut_16': False, 'mut_18': True, 'mut_20': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': False, 'mut_34': False, 'mut_36': False, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': True, 'mut_46': False, 'mut_48': False, 'mut_53': True, 'mut_56': False, 'mut_57': True, 'mut_59': True}. Best is trial 1 with value: -127.01452559072571.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1369771233 seed_offset=0 real_seed=1369771233\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1369771233 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633651 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1662850840792071513.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 55\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.186575 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1662850840792071513.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_5', 'mut_8', 'mut_12', 'mut_16', 'mut_22', 'mut_26', 'mut_28', 'mut_30', 'mut_33', 'mut_34', 'mut_36', 'mut_46', 'mut_48', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTTFSKEQLERLKEFNEKNRYPTAERREELASELGLNEAQVKVWFQNKRAKKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:24:51,388] Trial 2 finished with value: -53.57462593605695 and parameters: {'mut_1': False, 'mut_2': True, 'mut_5': False, 'mut_7': True, 'mut_8': False, 'mut_12': True, 'mut_13': False, 'mut_15': False, 'mut_16': True, 'mut_18': True, 'mut_20': True, 'mut_22': False, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': False, 'mut_31': True, 'mut_33': False, 'mut_34': True, 'mut_36': False, 'mut_37': False, 'mut_38': False, 'mut_42': True, 'mut_44': True, 'mut_46': True, 'mut_48': False, 'mut_53': False, 'mut_56': False, 'mut_57': True, 'mut_59': True}. Best is trial 1 with value: -127.01452559072571.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.46s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1282395387 seed_offset=0 real_seed=-1282395387\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1282395387 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.632436 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3129200185711932567.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 55\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.18808 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3129200185711932567.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_2', 'mut_8', 'mut_12', 'mut_15', 'mut_20', 'mut_22', 'mut_28', 'mut_31', 'mut_33', 'mut_44', 'mut_46', 'mut_53', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRKRTAFSKEQLERLEREFEKNRYLTAEREREQLSSELGLNERQVKIWFQNRRAKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:24:55,900] Trial 3 finished with value: -136.552454833756 and parameters: {'mut_1': False, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': True, 'mut_12': True, 'mut_13': False, 'mut_15': False, 'mut_16': True, 'mut_18': False, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_30': True, 'mut_31': False, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': False, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': True, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': False, 'mut_59': False}. Best is trial 3 with value: -136.552454833756.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.19s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-609906153 seed_offset=0 real_seed=-609906153\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-609906153 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633856 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6053620633241480155.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187509 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6053620633241480155.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_12', 'mut_15', 'mut_18', 'mut_20', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_42', 'mut_44', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLERLERELFENRYLSAEEREELSKELGLTERQIKIWFQNKRAEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:00,286] Trial 4 finished with value: -122.36749440179989 and parameters: {'mut_1': True, 'mut_2': True, 'mut_5': False, 'mut_7': False, 'mut_8': True, 'mut_12': True, 'mut_13': False, 'mut_15': True, 'mut_16': False, 'mut_18': False, 'mut_20': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_30': False, 'mut_31': True, 'mut_33': True, 'mut_34': False, 'mut_36': False, 'mut_37': False, 'mut_38': False, 'mut_42': False, 'mut_44': True, 'mut_46': True, 'mut_48': False, 'mut_53': True, 'mut_56': False, 'mut_57': False, 'mut_59': True}. Best is trial 3 with value: -136.552454833756.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-405538789 seed_offset=0 real_seed=-405538789\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-405538789 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.63402 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1636558675506846982.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187676 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1636558675506846982.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_18', 'mut_20', 'mut_22', 'mut_31', 'mut_33', 'mut_42', 'mut_48', 'mut_53']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTTFTKEQLAELKRELFEKNRYLTEREREQLSSELGLTEAQIKVWFQNRRAKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:04,666] Trial 5 finished with value: -131.31214127663637 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': True, 'mut_13': False, 'mut_15': True, 'mut_16': False, 'mut_18': True, 'mut_20': True, 'mut_22': False, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': True, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': False, 'mut_57': True, 'mut_59': False}. Best is trial 3 with value: -136.552454833756.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.47s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-309751842 seed_offset=0 real_seed=-309751842\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-309751842 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633318 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2277079140807618814.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 55\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187225 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2277079140807618814.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_2', 'mut_7', 'mut_12', 'mut_13', 'mut_15', 'mut_16', 'mut_22', 'mut_26', 'mut_28', 'mut_36', 'mut_42', 'mut_44', 'mut_48', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RKRTAFTSEQLEELEEFNEKNRYPTAERRRQQLASELGLTERQIKVWFQNKRAEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:09,229] Trial 6 finished with value: -107.22631875801795 and parameters: {'mut_1': False, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': False, 'mut_16': False, 'mut_18': True, 'mut_20': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_30': False, 'mut_31': True, 'mut_33': True, 'mut_34': False, 'mut_36': False, 'mut_37': False, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': True, 'mut_56': False, 'mut_57': False, 'mut_59': False}. Best is trial 3 with value: -136.552454833756.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.47s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1060942200 seed_offset=0 real_seed=1060942200\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1060942200 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633577 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1647397838259291818.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187715 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1647397838259291818.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:13,672] Trial 7 finished with value: -99.28168619291299 and parameters: {'mut_1': False, 'mut_2': True, 'mut_5': False, 'mut_7': True, 'mut_8': False, 'mut_12': True, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': False, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_30': False, 'mut_31': False, 'mut_33': False, 'mut_34': False, 'mut_36': True, 'mut_37': False, 'mut_38': False, 'mut_42': True, 'mut_44': True, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': False, 'mut_57': True, 'mut_59': True}. Best is trial 3 with value: -136.552454833756.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.77s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=55532658 seed_offset=0 real_seed=55532658\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=55532658 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634623 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187962 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_2', 'mut_5', 'mut_7', 'mut_13', 'mut_16', 'mut_18', 'mut_20', 'mut_22', 'mut_26', 'mut_36', 'mut_37', 'mut_42', 'mut_46', 'mut_53', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RKRTTFTSEQLAELKELFEKNRYPTERRRQQLAKELGLTEAQVKIWFQNRRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:18,485] Trial 8 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.20s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1500717260 seed_offset=0 real_seed=1500717260\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1500717260 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634115 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5086198126540980691.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188142 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5086198126540980691.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:22,692] Trial 9 finished with value: -144.19936671507972 and parameters: {'mut_1': False, 'mut_2': True, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': False, 'mut_16': True, 'mut_18': True, 'mut_20': True, 'mut_22': True, 'mut_26': True, 'mut_27': False, 'mut_28': False, 'mut_30': False, 'mut_31': False, 'mut_33': False, 'mut_34': False, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': True, 'mut_48': False, 'mut_53': True, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.27s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-417004730 seed_offset=0 real_seed=-417004730\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-417004730 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.63386 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188342 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_30', 'mut_36', 'mut_37', 'mut_42', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLAELEELFNEKNRYPTERRQQLAKELGLTEAQIKIWFQNKRAEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:27,105] Trial 10 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.47s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-616799875 seed_offset=0 real_seed=-616799875\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-616799875 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633844 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2130546133771866549.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.18958 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2130546133771866549.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_5', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_27', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTTFSKEQLAELEELFNEKNRYLSEEREELSKELGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:31,683] Trial 11 finished with value: -114.01830693963116 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': False, 'mut_28': False, 'mut_30': True, 'mut_31': False, 'mut_33': False, 'mut_34': False, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': False, 'mut_57': True, 'mut_59': True}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-7228146 seed_offset=0 real_seed=-7228146\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-7228146 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.63397 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-27362454334340756.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188814 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-27362454334340756.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_18', 'mut_20', 'mut_22', 'mut_31', 'mut_33', 'mut_42', 'mut_48', 'mut_53']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTTFTKEQLAELKRELFEKNRYLTEREREQLSSELGLTEAQIKVWFQNRRAKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:36,248] Trial 12 finished with value: -87.66405364215598 and parameters: {'mut_1': False, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': False, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.01s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-295252814 seed_offset=0 real_seed=-295252814\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-295252814 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634103 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2277079140807618814.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 55\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188523 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2277079140807618814.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_2', 'mut_7', 'mut_12', 'mut_13', 'mut_15', 'mut_18', 'mut_22', 'mut_28', 'mut_30', 'mut_37', 'mut_42', 'mut_44', 'mut_48', 'mut_53', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RKRTAFTSEQLEELERELFNEKNRYLTAERRQQLSKELGLTERQIKVWFQNRRAEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:40,362] Trial 13 finished with value: -107.22631875801795 and parameters: {'mut_1': False, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': False, 'mut_16': False, 'mut_18': True, 'mut_20': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_30': False, 'mut_31': True, 'mut_33': True, 'mut_34': False, 'mut_36': False, 'mut_37': False, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': True, 'mut_56': False, 'mut_57': False, 'mut_59': False}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.47s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1993135578 seed_offset=0 real_seed=-1993135578\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1993135578 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634141 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-909012586154063037.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 57\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187774 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-909012586154063037.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_31', 'mut_33', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_53', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLKELFNEKNRYPSEREREQLAKKLGLNEAQIKVWFQNRRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:45,291] Trial 14 finished with value: -65.34572651836143 and parameters: {'mut_1': False, 'mut_2': True, 'mut_5': False, 'mut_7': True, 'mut_8': False, 'mut_12': True, 'mut_13': True, 'mut_15': True, 'mut_16': False, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_30': True, 'mut_31': False, 'mut_33': False, 'mut_34': False, 'mut_36': False, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': True, 'mut_46': False, 'mut_48': True, 'mut_53': True, 'mut_56': False, 'mut_57': True, 'mut_59': True}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1665209977 seed_offset=0 real_seed=-1665209977\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1665209977 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634284 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3168230533244720392.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187261 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3168230533244720392.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_8', 'mut_12', 'mut_15', 'mut_22', 'mut_28', 'mut_30', 'mut_46', 'mut_56', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTAFSKEQLERLEREFNEKNRYLTAERRQQLSSELGLNEAQVKIWFQNKRAKKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:49,802] Trial 15 finished with value: -139.01289406651878 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': False, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_30': False, 'mut_31': True, 'mut_33': True, 'mut_34': False, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': True, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.19s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-50996936 seed_offset=0 real_seed=-50996936\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-50996936 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633163 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2979230296669084134.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 57\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.186953 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2979230296669084134.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_38', 'mut_44', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELSKKLGLNERQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:54,100] Trial 16 finished with value: -129.10656107409184 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': False, 'mut_7': False, 'mut_8': True, 'mut_12': True, 'mut_13': False, 'mut_15': True, 'mut_16': False, 'mut_18': False, 'mut_20': False, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_30': True, 'mut_31': False, 'mut_33': False, 'mut_34': False, 'mut_36': False, 'mut_37': False, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': True, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': False, 'mut_59': True}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.27s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=277378469 seed_offset=0 real_seed=277378469\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=277378469 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633583 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/7070142618387167484.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188225 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/7070142618387167484.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_2', 'mut_5', 'mut_12', 'mut_16', 'mut_18', 'mut_20', 'mut_26', 'mut_27', 'mut_28', 'mut_31', 'mut_33', 'mut_34', 'mut_44', 'mut_46', 'mut_48']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RKRTTFSSEQLERLKELFENRYPSAEREREELSSELGLNERQVKVWFQNKRAKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:25:58,595] Trial 17 finished with value: -147.91313961386913 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': True, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.48s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=70743727 seed_offset=0 real_seed=70743727\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=70743727 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.632965 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5122801370586612803.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 54\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187556 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5122801370586612803.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_15', 'mut_18', 'mut_20', 'mut_22', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_36', 'mut_38', 'mut_44', 'mut_53', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTAFSSEQLARLERELFEKNRYLSAEERQQLASKLGLNERQIKIWFQNRRAEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:03,201] Trial 18 finished with value: -122.34419698397282 and parameters: {'mut_1': False, 'mut_2': True, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': True, 'mut_13': False, 'mut_15': False, 'mut_16': True, 'mut_18': True, 'mut_20': True, 'mut_22': False, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': False, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': False, 'mut_38': False, 'mut_42': False, 'mut_44': True, 'mut_46': True, 'mut_48': True, 'mut_53': False, 'mut_56': False, 'mut_57': False, 'mut_59': False}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.20s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1094186745 seed_offset=0 real_seed=1094186745\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1094186745 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633361 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8808614615809036204.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187359 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8808614615809036204.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_5', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_31', 'mut_36', 'mut_37', 'mut_53', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTTFSSEQLAELEELFNEKNRYPSERERQQLAKELGLNEAQIKIWFQNRRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:07,365] Trial 19 finished with value: -102.57575400509666 and parameters: {'mut_1': False, 'mut_2': False, 'mut_5': False, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': False, 'mut_18': True, 'mut_20': True, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': False, 'mut_34': False, 'mut_36': True, 'mut_37': False, 'mut_38': True, 'mut_42': False, 'mut_44': True, 'mut_46': False, 'mut_48': False, 'mut_53': True, 'mut_56': False, 'mut_57': True, 'mut_59': False}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-269074686 seed_offset=0 real_seed=-269074686\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-269074686 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633674 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3882515369876252029.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 57\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188858 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3882515369876252029.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_12', 'mut_13', 'mut_16', 'mut_18', 'mut_20', 'mut_22', 'mut_26', 'mut_36', 'mut_37', 'mut_42', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLEELKELFEKNRYPTERRRQQLAKELGLTEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:11,588] Trial 20 finished with value: -146.66308410551392 and parameters: {'mut_1': False, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_30': False, 'mut_31': True, 'mut_33': False, 'mut_34': False, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': True, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.23s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=962107268 seed_offset=0 real_seed=962107268\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=962107268 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.632564 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3118574613826795600.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 57\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188055 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3118574613826795600.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:15,887] Trial 21 finished with value: -143.22213800976755 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': True, 'mut_13': True, 'mut_15': False, 'mut_16': True, 'mut_18': True, 'mut_20': True, 'mut_22': True, 'mut_26': True, 'mut_27': False, 'mut_28': False, 'mut_30': False, 'mut_31': False, 'mut_33': False, 'mut_34': False, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 8 with value: -149.21925008304527.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=102813888 seed_offset=0 real_seed=102813888\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=102813888 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.632519 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/38000268666189396.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.189547 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/38000268666189396.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_5', 'mut_7', 'mut_13', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_31', 'mut_33', 'mut_36', 'mut_37', 'mut_53', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTTFTSEQLAELKELFNEKNRYPSAEREREQLAKELGLNEAQIKIWFQNRRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:20,178] Trial 22 finished with value: -149.67542513173834 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 22 with value: -149.67542513173834.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1898052548 seed_offset=0 real_seed=-1898052548\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1898052548 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633422 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-8215452895843666495.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187853 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-8215452895843666495.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:24,619] Trial 23 finished with value: -98.28484757065027 and parameters: {'mut_1': False, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': False, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': False, 'mut_31': True, 'mut_33': True, 'mut_34': False, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': True, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 22 with value: -149.67542513173834.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1651613395 seed_offset=0 real_seed=1651613395\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1651613395 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633245 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187411 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_38', 'mut_44', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELSKKLGLNERQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:28,932] Trial 24 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 22 with value: -149.67542513173834.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.02it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-153280051 seed_offset=0 real_seed=-153280051\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-153280051 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633404 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/7070142618387167484.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.186992 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/7070142618387167484.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_7', 'mut_8', 'mut_12', 'mut_15', 'mut_16', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_38', 'mut_44', 'mut_56', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTAFTKEQLERLEEFNENRYLSAEERQQLSSKLGLNERQIKIWFQNKRAKKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:33,137] Trial 25 finished with value: -147.91313961386913 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': True, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 22 with value: -149.67542513173834.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1237298026 seed_offset=0 real_seed=-1237298026\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1237298026 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633223 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8964999217039165899.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 55\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188653 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8964999217039165899.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_12', 'mut_15', 'mut_18', 'mut_20', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_42', 'mut_44', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLERLERELFENRYLSAEEREELSKELGLTERQIKIWFQNKRAEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:37,504] Trial 26 finished with value: -122.21752294525278 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': False, 'mut_7': True, 'mut_8': True, 'mut_12': True, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': False, 'mut_20': False, 'mut_22': False, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': False, 'mut_34': False, 'mut_36': False, 'mut_37': False, 'mut_38': True, 'mut_42': False, 'mut_44': True, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': False, 'mut_59': True}. Best is trial 22 with value: -149.67542513173834.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.02it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-85781911 seed_offset=0 real_seed=-85781911\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-85781911 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633845 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1636558675506846982.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187364 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1636558675506846982.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:41,688] Trial 27 finished with value: -131.31214127663637 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': True, 'mut_13': False, 'mut_15': True, 'mut_16': False, 'mut_18': True, 'mut_20': True, 'mut_22': False, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': True, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': False, 'mut_57': True, 'mut_59': False}. Best is trial 22 with value: -149.67542513173834.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.00it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=687277948 seed_offset=0 real_seed=687277948\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=687277948 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.63356 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187262 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_22', 'mut_26', 'mut_28', 'mut_30', 'mut_31', 'mut_34', 'mut_38', 'mut_46', 'mut_48', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEREFNEKNRYPTAEERQELSSKLGLNEAQVKVWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:45,748] Trial 28 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 22 with value: -149.67542513173834.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.45s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1499363448 seed_offset=0 real_seed=1499363448\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1499363448 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633483 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5750848353176039570.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187935 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5750848353176039570.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_5', 'mut_15', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_30', 'mut_31', 'mut_33', 'mut_36', 'mut_37', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTTFSSEQLARLERELFNEKNRYPSEEREQLAKELGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:50,310] Trial 29 finished with value: -143.85367826521758 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': False, 'mut_18': False, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': False, 'mut_34': True, 'mut_36': False, 'mut_37': False, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': True, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 22 with value: -149.67542513173834.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.46s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=984122353 seed_offset=0 real_seed=984122353\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=984122353 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634109 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2102128262953428041.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 57\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188095 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2102128262953428041.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEREREELAKELGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:55,332] Trial 30 finished with value: -120.95699761204006 and parameters: {'mut_1': False, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': False, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': False, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 22 with value: -149.67542513173834.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.48s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1365154617 seed_offset=0 real_seed=-1365154617\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1365154617 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634247 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3506370479318070987.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 59\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.186754 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3506370479318070987.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_38', 'mut_44', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELSKKLGLNERQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:26:59,959] Trial 31 finished with value: -126.23214004002057 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': False, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 22 with value: -149.67542513173834.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1503101784 seed_offset=0 real_seed=1503101784\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1503101784 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633661 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8245284332749715868.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187243 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8245284332749715868.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLAELEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:04,285] Trial 32 finished with value: -149.89593876601836 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': True, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 32 with value: -149.89593876601836.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=918106265 seed_offset=0 real_seed=918106265\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=918106265 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633117 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/846075718137088726.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.186584 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/846075718137088726.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:08,592] Trial 33 finished with value: -151.03106968167893 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.00s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1668655923 seed_offset=0 real_seed=-1668655923\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1668655923 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633725 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187396 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:12,633] Trial 34 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.02it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=2020088236 seed_offset=0 real_seed=2020088236\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=2020088236 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.632889 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.189002 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:16,683] Trial 35 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.29s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1814294376 seed_offset=0 real_seed=1814294376\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1814294376 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.632733 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.1892 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:21,016] Trial 36 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=35552596 seed_offset=0 real_seed=35552596\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=35552596 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633418 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188422 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_53', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSEEREELAKKLGLNEAQIKIWFQNRRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:25,313] Trial 37 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.20s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1687739881 seed_offset=0 real_seed=-1687739881\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1687739881 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633251 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6599468276360441063.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 57\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188204 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6599468276360441063.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_20', 'mut_22', 'mut_26', 'mut_30', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLAELEELFEKNRYPTERREELAKKLGLNEAQIKVWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:29,753] Trial 38 finished with value: -131.83353722722703 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': True, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.46s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1831456553 seed_offset=0 real_seed=-1831456553\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1831456553 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633096 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7120449639603835376.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187744 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7120449639603835376.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:34,228] Trial 39 finished with value: -139.8991257138298 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': True, 'mut_22': True, 'mut_26': True, 'mut_27': False, 'mut_28': False, 'mut_30': True, 'mut_31': False, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-80751766 seed_offset=0 real_seed=-80751766\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-80751766 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633581 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5467499534871976069.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.189172 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5467499534871976069.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:38,487] Trial 40 finished with value: -148.88080093411654 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.01s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-962935337 seed_offset=0 real_seed=-962935337\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-962935337 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.632962 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.18839 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:42,534] Trial 41 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.28s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=904429389 seed_offset=0 real_seed=904429389\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=904429389 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633054 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/38000268666189396.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188334 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/38000268666189396.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_44', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSKEQLARLEELFNEKNRYPSAEEREELAKKLGLNERQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:46,846] Trial 42 finished with value: -149.67542513173834 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1379349437 seed_offset=0 real_seed=1379349437\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1379349437 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.63352 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1874959681574331955.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.186919 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1874959681574331955.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:51,132] Trial 43 finished with value: -150.91621511555974 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': True, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.02it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-305958070 seed_offset=0 real_seed=-305958070\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-305958070 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634053 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187905 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:55,164] Trial 44 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.01it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1502158351 seed_offset=0 real_seed=-1502158351\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1502158351 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633344 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188403 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3749267196841071361.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_38', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLAELEELFNEKNRYPSAEEREELSKKLGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:27:59,200] Trial 45 finished with value: -149.21925008304527 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.01it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1904896084 seed_offset=0 real_seed=-1904896084\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1904896084 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633671 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-227336056950384135.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188373 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-227336056950384135.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_38', 'mut_44', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLAELEELFNEKNRYPSAEEREELSKKLGLNERQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:03,327] Trial 46 finished with value: -149.0146873459708 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1068395650 seed_offset=0 real_seed=-1068395650\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1068395650 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634031 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-8483440683322439762.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188395 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-8483440683322439762.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLAELEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:07,604] Trial 47 finished with value: -144.94125941532693 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': True, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.45s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1639050414 seed_offset=0 real_seed=1639050414\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1639050414 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634199 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6900068507267422134.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188066 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6900068507267422134.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:12,216] Trial 48 finished with value: -150.62017065239363 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.29s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-932946406 seed_offset=0 real_seed=-932946406\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-932946406 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633378 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/38000268666189396.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187662 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/38000268666189396.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_44', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLAELEELFNEKNRYPSAEEREELAKELGLNERQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:16,545] Trial 49 finished with value: -149.67542513173834 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=568253688 seed_offset=0 real_seed=568253688\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=568253688 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633666 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7069923152269188853.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187957 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7069923152269188853.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_42', 'mut_46', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLAELEELFNEKNRYPSAEEREELAKELGLTEAQVKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:20,878] Trial 50 finished with value: -146.27306287634426 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': True, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 33 with value: -151.03106968167893.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1146393874 seed_offset=0 real_seed=-1146393874\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1146393874 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634876 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3095664635564171363.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187495 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3095664635564171363.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:25,152] Trial 51 finished with value: -151.64605111519947 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': True, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 51 with value: -151.64605111519947.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.46s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-294717552 seed_offset=0 real_seed=-294717552\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-294717552 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.632673 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-9005758357727744758.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188058 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-9005758357727744758.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLARLEELFNEKNRYPSAERREELAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:29,700] Trial 52 finished with value: -149.8602314069566 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 51 with value: -151.64605111519947.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.46s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=273336552 seed_offset=0 real_seed=273336552\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=273336552 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633562 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6756424967688718581.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188017 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6756424967688718581.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLARLEELFNEKNRYPSAEEREELSKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:34,335] Trial 53 finished with value: -144.45544993954795 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': False, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 51 with value: -151.64605111519947.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.44s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1639675036 seed_offset=0 real_seed=-1639675036\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1639675036 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633262 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2812641522224101340.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187117 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2812641522224101340.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLAELEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:39,051] Trial 54 finished with value: -148.54205957761965 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 51 with value: -151.64605111519947.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.29s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-628328978 seed_offset=0 real_seed=-628328978\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-628328978 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.63352 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/846075718137088726.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187767 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/846075718137088726.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_36', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLARLEELFNEKNRYPSAEEREQLAKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:43,488] Trial 55 finished with value: -151.03106968167893 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 51 with value: -151.64605111519947.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.19s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=897253932 seed_offset=0 real_seed=897253932\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=897253932 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633165 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4536496856358284158.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187269 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4536496856358284158.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_38', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELSKKLGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:47,763] Trial 56 finished with value: -148.75335362247273 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': False, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 51 with value: -151.64605111519947.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.19s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-832601146 seed_offset=0 real_seed=-832601146\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-832601146 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634122 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-645702098858906778.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187097 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-645702098858906778.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:52,192] Trial 57 finished with value: -149.66256249869795 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 51 with value: -151.64605111519947.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1436356704 seed_offset=0 real_seed=1436356704\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1436356704 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633596 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.18776 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSSEQLAELEELFNEKNRYLSAEEREELSKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:28:56,468] Trial 58 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.46s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1447100641 seed_offset=0 real_seed=1447100641\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1447100641 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633737 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1660147564775233115.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187109 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1660147564775233115.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLAELEELFNEKNRYPSAEREREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:01,106] Trial 59 finished with value: -133.4968545808923 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.45s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1164099896 seed_offset=0 real_seed=-1164099896\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1164099896 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634234 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2354858578176341114.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 59\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187607 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2354858578176341114.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLAELEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:05,389] Trial 60 finished with value: 16211.89619006716 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': False, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-679318692 seed_offset=0 real_seed=-679318692\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-679318692 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633323 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6900068507267422134.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187285 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6900068507267422134.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:09,816] Trial 61 finished with value: -150.62017065239363 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.31s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1072109052 seed_offset=0 real_seed=1072109052\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1072109052 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633864 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187883 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_2', 'mut_5', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRKRTTFSKEQLARLEELFNEKNRYPSAERREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:14,246] Trial 62 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-754624561 seed_offset=0 real_seed=-754624561\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-754624561 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.635117 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2807536135918385513.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187271 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2807536135918385513.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:18,602] Trial 63 finished with value: -148.62448831400138 and parameters: {'mut_1': True, 'mut_2': True, 'mut_5': True, 'mut_7': False, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': False, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.01it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-600522749 seed_offset=0 real_seed=-600522749\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-600522749 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634235 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-9005758357727744758.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.186999 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-9005758357727744758.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_42', 'mut_46', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLAELEELFNEKNRYPSAEEREELAKELGLTEAQVKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:22,675] Trial 64 finished with value: -149.8602314069566 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.01it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=257533007 seed_offset=0 real_seed=257533007\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=257533007 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633709 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3095664635564171363.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187173 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3095664635564171363.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKKLGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:26,767] Trial 65 finished with value: -151.64605111519947 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': True, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.44s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1595988836 seed_offset=0 real_seed=-1595988836\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1595988836 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633535 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1054194420226214891.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188795 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1054194420226214891.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_37', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLAELEELFNEKNRYPSAEEREELSKELGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:31,342] Trial 66 finished with value: -151.02090466861117 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.19s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1871619545 seed_offset=0 real_seed=1871619545\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1871619545 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634376 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1636788963984161320.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188324 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1636788963984161320.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_48', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLAELEELFNEKNRYPSAEEREELAKELGLNEAQIKVWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:35,648] Trial 67 finished with value: -146.85022235982834 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': False, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.45s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1577065141 seed_offset=0 real_seed=-1577065141\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1577065141 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634093 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/4656970778367265338.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.1891 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/4656970778367265338.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:40,186] Trial 68 finished with value: -148.1363102743526 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': True, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.46s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1304736121 seed_offset=0 real_seed=1304736121\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1304736121 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.63366 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6592672011331995074.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188641 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6592672011331995074.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:44,754] Trial 69 finished with value: -153.67684385591878 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.02it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-263210877 seed_offset=0 real_seed=-263210877\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-263210877 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634202 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188262 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_42', 'mut_46', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLAELEELFNEKNRYPSAEEREELAKELGLTEAQVKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:48,850] Trial 70 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1474521525 seed_offset=0 real_seed=1474521525\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1474521525 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633503 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3095664635564171363.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.18716 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3095664635564171363.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:53,226] Trial 71 finished with value: -151.64605111519947 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': True, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-657469150 seed_offset=0 real_seed=-657469150\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-657469150 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633801 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.1876 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSEEREELAKELGLNEAQIKIWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:29:57,617] Trial 72 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.02it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=751178999 seed_offset=0 real_seed=751178999\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=751178999 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634253 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/887128745649416405.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 57\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187075 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/887128745649416405.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLAELEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:01,788] Trial 73 finished with value: -132.09452829101932 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-2035641630 seed_offset=0 real_seed=-2035641630\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-2035641630 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633404 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/846075718137088726.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187957 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/846075718137088726.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_42', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLAELEELFNEKNRYPSAEEREELAKELGLTEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:06,194] Trial 74 finished with value: -151.03106968167893 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.44s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1149497245 seed_offset=0 real_seed=-1149497245\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1149497245 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634535 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8938643486610584963.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188227 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8938643486610584963.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:10,744] Trial 75 finished with value: -150.0452641813104 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1938675768 seed_offset=0 real_seed=-1938675768\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1938675768 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.63358 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6592672011331995074.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188693 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6592672011331995074.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_34', 'mut_36', 'mut_37', 'mut_42', 'mut_46', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFSKEQLAELEELFNEKNRYPSAEERQELAKELGLTEAQVKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:15,117] Trial 76 finished with value: -153.67684385591878 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1938581113 seed_offset=0 real_seed=1938581113\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1938581113 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634457 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/7077416504878814730.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188466 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/7077416504878814730.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_13', 'mut_15', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTSEQLAELERELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:19,335] Trial 77 finished with value: -150.95526369849202 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': False, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': False, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': True, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.21s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1933074584 seed_offset=0 real_seed=1933074584\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1933074584 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634062 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3218854954866424191.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 59\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187358 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3218854954866424191.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_42', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLAELEELFNEKNRYPSAEEREELAKELGLTEAQIKIWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:24,044] Trial 78 finished with value: -141.66791367749218 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': False, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.45s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1444000733 seed_offset=0 real_seed=1444000733\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1444000733 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633982 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-897428722434339754.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187819 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-897428722434339754.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:28,581] Trial 79 finished with value: -146.84636760454433 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1338180556 seed_offset=0 real_seed=1338180556\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1338180556 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633885 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187712 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:32,989] Trial 80 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.01s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-170074913 seed_offset=0 real_seed=-170074913\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-170074913 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634012 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6592672011331995074.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187954 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6592672011331995074.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_7', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_46', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTAFTSEQLAELEELFNEKNRYPSAERREELAKELGLNEAQVKIWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:37,108] Trial 81 finished with value: -153.67684385591878 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-2091511330 seed_offset=0 real_seed=-2091511330\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-2091511330 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634742 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6271914762934038129.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187887 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6271914762934038129.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_7', 'mut_8', 'mut_12', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTAFTKEQLEELEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:41,436] Trial 82 finished with value: -144.13991830820683 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': False, 'mut_7': True, 'mut_8': False, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': False, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': True, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.51s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1732016021 seed_offset=0 real_seed=1732016021\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1732016021 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633712 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/4165805602703424122.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187924 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/4165805602703424122.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:45,946] Trial 83 finished with value: -148.07474470537795 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': False, 'mut_7': True, 'mut_8': True, 'mut_12': True, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-554397128 seed_offset=0 real_seed=-554397128\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-554397128 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633426 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6592672011331995074.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188655 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6592672011331995074.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLERELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:50,316] Trial 84 finished with value: -153.67684385591878 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.47s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-319987401 seed_offset=0 real_seed=-319987401\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-319987401 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633784 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6762173505900724949.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 59\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188345 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6762173505900724949.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:54,971] Trial 85 finished with value: -136.7596618529335 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': False, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.01it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=205881959 seed_offset=0 real_seed=205881959\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=205881959 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634161 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6592672011331995074.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188612 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6592672011331995074.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:30:59,068] Trial 86 finished with value: -153.67684385591878 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=842370523 seed_offset=0 real_seed=842370523\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=842370523 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634012 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187989 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:03,432] Trial 87 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1769861280 seed_offset=0 real_seed=1769861280\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1769861280 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633598 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188953 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_5', 'mut_7', 'mut_8', 'mut_13', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_42', 'mut_46', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTTFTKEQLAELEELFNEKNRYPSAEEREELAKELGLTEAQVKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:07,801] Trial 88 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.19s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1434435291 seed_offset=0 real_seed=-1434435291\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1434435291 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633549 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1782005080626969929.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 57\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.189061 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1782005080626969929.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:12,004] Trial 89 finished with value: -149.38771758887478 and parameters: {'mut_1': False, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': True, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': True, 'mut_44': False, 'mut_46': True, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.01it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-331473869 seed_offset=0 real_seed=-331473869\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-331473869 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634175 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.189095 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_12', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_38', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLERLEELFNEKNRYPSAEEREELAKKLGLNEAQIKIWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:16,118] Trial 90 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 58 with value: -154.7812656192612.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.55s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1438648193 seed_offset=0 real_seed=1438648193\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1438648193 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633304 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1889697353087351511.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.189174 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1889697353087351511.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:20,801] Trial 91 finished with value: -155.10935372311928 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': True, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': True, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 91 with value: -155.10935372311928.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.03it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1438588478 seed_offset=0 real_seed=1438588478\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1438588478 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633385 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188218 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:24,891] Trial 92 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 91 with value: -155.10935372311928.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.03it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1505507164 seed_offset=0 real_seed=-1505507164\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1505507164 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633485 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188012 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57', 'mut_59']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLKELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKK\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:28,986] Trial 93 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 91 with value: -155.10935372311928.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-170382032 seed_offset=0 real_seed=-170382032\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-170382032 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633018 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5183936650968702885.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue LYS:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187951 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5183936650968702885.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:33,284] Trial 94 finished with value: -149.1543773149982 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': False, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': True}. Best is trial 91 with value: -155.10935372311928.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1330767954 seed_offset=0 real_seed=1330767954\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1330767954 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633838 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187476 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:37,654] Trial 95 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 91 with value: -155.10935372311928.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1529471627 seed_offset=0 real_seed=1529471627\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1529471627 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634167 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188439 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:42,022] Trial 96 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 91 with value: -155.10935372311928.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=578373839 seed_offset=0 real_seed=578373839\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=578373839 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.634613 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187499 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:46,397] Trial 97 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 91 with value: -155.10935372311928.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.03it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-444071469 seed_offset=0 real_seed=-444071469\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-444071469 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633526 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.188459 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_1', 'mut_5', 'mut_7', 'mut_8', 'mut_15', 'mut_16', 'mut_18', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_30', 'mut_31', 'mut_33', 'mut_34', 'mut_36', 'mut_37', 'mut_56', 'mut_57']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RRPRTTFTKEQLARLEELFNEKNRYPSAEEREELAKELGLNEAQIKIWFQNKRAKEKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:50,505] Trial 98 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 91 with value: -155.10935372311928.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.02it/s]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-566980697 seed_offset=0 real_seed=-566980697\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-566980697 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633807 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.187748 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/524605574788680739.pdb with minimized positions\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-01 12:31:54,599] Trial 99 finished with value: -154.7812656192612 and parameters: {'mut_1': True, 'mut_2': False, 'mut_5': True, 'mut_7': True, 'mut_8': True, 'mut_12': False, 'mut_13': False, 'mut_15': True, 'mut_16': True, 'mut_18': True, 'mut_20': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_30': True, 'mut_31': True, 'mut_33': True, 'mut_34': True, 'mut_36': True, 'mut_37': True, 'mut_38': False, 'mut_42': False, 'mut_44': False, 'mut_46': False, 'mut_48': False, 'mut_53': False, 'mut_56': True, 'mut_57': True, 'mut_59': False}. Best is trial 91 with value: -155.10935372311928.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "({'mut_1': True,\n",
       "  'mut_2': False,\n",
       "  'mut_5': True,\n",
       "  'mut_7': True,\n",
       "  'mut_8': True,\n",
       "  'mut_12': True,\n",
       "  'mut_13': False,\n",
       "  'mut_15': True,\n",
       "  'mut_16': True,\n",
       "  'mut_18': True,\n",
       "  'mut_20': False,\n",
       "  'mut_22': True,\n",
       "  'mut_26': True,\n",
       "  'mut_27': True,\n",
       "  'mut_28': True,\n",
       "  'mut_30': True,\n",
       "  'mut_31': True,\n",
       "  'mut_33': True,\n",
       "  'mut_34': True,\n",
       "  'mut_36': True,\n",
       "  'mut_37': True,\n",
       "  'mut_38': True,\n",
       "  'mut_42': False,\n",
       "  'mut_44': False,\n",
       "  'mut_46': False,\n",
       "  'mut_48': False,\n",
       "  'mut_53': False,\n",
       "  'mut_56': True,\n",
       "  'mut_57': True,\n",
       "  'mut_59': True},\n",
       " -155.10935372311928,\n",
       " <optuna.study.study.Study at 0x14b928cea040>)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "optimizer.run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "85e193c3-9859-4994-b39a-575ca5deb150",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>number</th>\n",
       "      <th>value</th>\n",
       "      <th>datetime_start</th>\n",
       "      <th>datetime_complete</th>\n",
       "      <th>duration</th>\n",
       "      <th>params_mut_1</th>\n",
       "      <th>params_mut_12</th>\n",
       "      <th>params_mut_13</th>\n",
       "      <th>params_mut_15</th>\n",
       "      <th>params_mut_16</th>\n",
       "      <th>...</th>\n",
       "      <th>params_mut_48</th>\n",
       "      <th>params_mut_5</th>\n",
       "      <th>params_mut_53</th>\n",
       "      <th>params_mut_56</th>\n",
       "      <th>params_mut_57</th>\n",
       "      <th>params_mut_59</th>\n",
       "      <th>params_mut_7</th>\n",
       "      <th>params_mut_8</th>\n",
       "      <th>system_attrs_nsga2:generation</th>\n",
       "      <th>state</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>-118.651758</td>\n",
       "      <td>2023-08-01 12:24:07.284748</td>\n",
       "      <td>2023-08-01 12:24:42.501752</td>\n",
       "      <td>0 days 00:00:35.217004</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>COMPLETE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>-127.014526</td>\n",
       "      <td>2023-08-01 12:24:42.505533</td>\n",
       "      <td>2023-08-01 12:24:47.337485</td>\n",
       "      <td>0 days 00:00:04.831952</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>COMPLETE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>-53.574626</td>\n",
       "      <td>2023-08-01 12:24:47.341044</td>\n",
       "      <td>2023-08-01 12:24:51.387568</td>\n",
       "      <td>0 days 00:00:04.046524</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>COMPLETE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>-136.552455</td>\n",
       "      <td>2023-08-01 12:24:51.390991</td>\n",
       "      <td>2023-08-01 12:24:55.899605</td>\n",
       "      <td>0 days 00:00:04.508614</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>COMPLETE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>-122.367494</td>\n",
       "      <td>2023-08-01 12:24:55.903087</td>\n",
       "      <td>2023-08-01 12:25:00.286195</td>\n",
       "      <td>0 days 00:00:04.383108</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>COMPLETE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>95</td>\n",
       "      <td>-154.781266</td>\n",
       "      <td>2023-08-01 12:31:33.286943</td>\n",
       "      <td>2023-08-01 12:31:37.654102</td>\n",
       "      <td>0 days 00:00:04.367159</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>9</td>\n",
       "      <td>COMPLETE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>96</td>\n",
       "      <td>-154.781266</td>\n",
       "      <td>2023-08-01 12:31:37.657381</td>\n",
       "      <td>2023-08-01 12:31:42.021948</td>\n",
       "      <td>0 days 00:00:04.364567</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>9</td>\n",
       "      <td>COMPLETE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>97</td>\n",
       "      <td>-154.781266</td>\n",
       "      <td>2023-08-01 12:31:42.025390</td>\n",
       "      <td>2023-08-01 12:31:46.394417</td>\n",
       "      <td>0 days 00:00:04.369027</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>9</td>\n",
       "      <td>COMPLETE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>98</td>\n",
       "      <td>-154.781266</td>\n",
       "      <td>2023-08-01 12:31:46.399871</td>\n",
       "      <td>2023-08-01 12:31:50.504547</td>\n",
       "      <td>0 days 00:00:04.104676</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>9</td>\n",
       "      <td>COMPLETE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>99</td>\n",
       "      <td>-154.781266</td>\n",
       "      <td>2023-08-01 12:31:50.508039</td>\n",
       "      <td>2023-08-01 12:31:54.598387</td>\n",
       "      <td>0 days 00:00:04.090348</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>9</td>\n",
       "      <td>COMPLETE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    number       value             datetime_start          datetime_complete  \\\n",
       "0        0 -118.651758 2023-08-01 12:24:07.284748 2023-08-01 12:24:42.501752   \n",
       "1        1 -127.014526 2023-08-01 12:24:42.505533 2023-08-01 12:24:47.337485   \n",
       "2        2  -53.574626 2023-08-01 12:24:47.341044 2023-08-01 12:24:51.387568   \n",
       "3        3 -136.552455 2023-08-01 12:24:51.390991 2023-08-01 12:24:55.899605   \n",
       "4        4 -122.367494 2023-08-01 12:24:55.903087 2023-08-01 12:25:00.286195   \n",
       "..     ...         ...                        ...                        ...   \n",
       "95      95 -154.781266 2023-08-01 12:31:33.286943 2023-08-01 12:31:37.654102   \n",
       "96      96 -154.781266 2023-08-01 12:31:37.657381 2023-08-01 12:31:42.021948   \n",
       "97      97 -154.781266 2023-08-01 12:31:42.025390 2023-08-01 12:31:46.394417   \n",
       "98      98 -154.781266 2023-08-01 12:31:46.399871 2023-08-01 12:31:50.504547   \n",
       "99      99 -154.781266 2023-08-01 12:31:50.508039 2023-08-01 12:31:54.598387   \n",
       "\n",
       "                 duration  params_mut_1  params_mut_12  params_mut_13  \\\n",
       "0  0 days 00:00:35.217004          True          False          False   \n",
       "1  0 days 00:00:04.831952         False          False           True   \n",
       "2  0 days 00:00:04.046524         False           True          False   \n",
       "3  0 days 00:00:04.508614         False           True          False   \n",
       "4  0 days 00:00:04.383108          True           True          False   \n",
       "..                    ...           ...            ...            ...   \n",
       "95 0 days 00:00:04.367159          True          False          False   \n",
       "96 0 days 00:00:04.364567          True          False          False   \n",
       "97 0 days 00:00:04.369027          True          False          False   \n",
       "98 0 days 00:00:04.104676          True          False          False   \n",
       "99 0 days 00:00:04.090348          True          False          False   \n",
       "\n",
       "    params_mut_15  params_mut_16  ...  params_mut_48  params_mut_5  \\\n",
       "0           False          False  ...           True          True   \n",
       "1           False          False  ...          False         False   \n",
       "2           False           True  ...          False         False   \n",
       "3           False           True  ...           True          True   \n",
       "4            True          False  ...          False         False   \n",
       "..            ...            ...  ...            ...           ...   \n",
       "95           True           True  ...          False          True   \n",
       "96           True           True  ...          False          True   \n",
       "97           True           True  ...          False          True   \n",
       "98           True           True  ...          False          True   \n",
       "99           True           True  ...          False          True   \n",
       "\n",
       "    params_mut_53  params_mut_56  params_mut_57  params_mut_59  params_mut_7  \\\n",
       "0           False           True           True           True          True   \n",
       "1            True          False           True           True          True   \n",
       "2           False          False           True           True          True   \n",
       "3           False           True          False          False         False   \n",
       "4            True          False          False           True         False   \n",
       "..            ...            ...            ...            ...           ...   \n",
       "95          False           True           True          False          True   \n",
       "96          False           True           True          False          True   \n",
       "97          False           True           True          False          True   \n",
       "98          False           True           True          False          True   \n",
       "99          False           True           True          False          True   \n",
       "\n",
       "    params_mut_8  system_attrs_nsga2:generation     state  \n",
       "0          False                              0  COMPLETE  \n",
       "1          False                              0  COMPLETE  \n",
       "2          False                              0  COMPLETE  \n",
       "3           True                              0  COMPLETE  \n",
       "4           True                              0  COMPLETE  \n",
       "..           ...                            ...       ...  \n",
       "95          True                              9  COMPLETE  \n",
       "96          True                              9  COMPLETE  \n",
       "97          True                              9  COMPLETE  \n",
       "98          True                              9  COMPLETE  \n",
       "99          True                              9  COMPLETE  \n",
       "\n",
       "[100 rows x 37 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "optimizer.study.trials_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "3e60c03c-57ff-4455-ae4d-ab82eab9a942",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'RRPRTTFTKEQLERLEELFNEKNRYPSAEEREELAKKLGLNEAQIKIWFQNKRAKEKK'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "optimizer.optimized_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3b3b536d-f952-4716-b10b-0d1823866b58",
   "metadata": {},
   "outputs": [],
   "source": [
    "best_file_hash = optimizer.hash_mutation_set(optimizer.best_mutations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "77b6151f-b1b5-4866-a69c-040aae2c8929",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-1889697353087351511"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "best_file_hash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "5b76e921-21f1-4422-b587-79f6c8c74582",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.rename(f'./tmp/mutation_optimizer_run/{best_file_hash}.pdb', f'./tmp/mutation_optimizer_run/consensus_optimized.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5c2cb786-9a6a-4336-a89c-338b2323d076",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2e06734660534e629486f56548f2eb40",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": []
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from Bio.PDB import *\n",
    "import nglview as nv\n",
    "import ipywidgets\n",
    "parser = PDBParser()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3644c763-143e-40c4-a9c0-bd63b7c2a053",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '# All ' at line 989\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '#BEGIN' at line 990\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'label ' at line 991\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'weight' at line 992\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'pose -' at line 993\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG:Nt' at line 994\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PRO_2 ' at line 995\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_3 ' at line 996\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_4 ' at line 997\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_5 ' at line 998\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_6 ' at line 999\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_7 ' at line 1000\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_8 ' at line 1001\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_9 ' at line 1002\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_10' at line 1003\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_11' at line 1004\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_12' at line 1005\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_13' at line 1006\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_14' at line 1007\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_15' at line 1008\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_16' at line 1009\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_17' at line 1010\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_18' at line 1011\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_19' at line 1012\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_20' at line 1013\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_21' at line 1014\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_22' at line 1015\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TYR_23' at line 1016\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PRO_24' at line 1017\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_25' at line 1018\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_26' at line 1019\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_27' at line 1020\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_28' at line 1021\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_29' at line 1022\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_30' at line 1023\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_31' at line 1024\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_32' at line 1025\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_33' at line 1026\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_34' at line 1027\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_35' at line 1028\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_36' at line 1029\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLY_37' at line 1030\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_38' at line 1031\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_39' at line 1032\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_40' at line 1033\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_41' at line 1034\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_42' at line 1035\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'VAL_43' at line 1036\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_44' at line 1037\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_45' at line 1038\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TRP_46' at line 1039\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_47' at line 1040\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_48' at line 1041\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_49' at line 1042\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_50' at line 1043\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_51' at line 1044\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_52' at line 1045\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_53' at line 1046\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_54' at line 1047\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_55' at line 1048\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS:Ct' at line 1049\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '#END_P' at line 1050\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "struct = parser.get_structure('opt', f'./tmp/ESMFoldDGEstimator_structures/{best_file_hash}.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "4d9de564-21de-48ff-ac1c-c9c803870fa2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d58ebef32317461e9127a2bfe3a3bb8e",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "NGLWidget()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "ab432a9be7624a429a2b544647fbdaf5",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Tab(children=(Box(children=(Box(children=(Box(children=(Label(value='step'), IntSlider(value=1, min=-100)), la…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nv.show_biopython(struct, gui=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "027c55f6-291b-4382-9853-a3b98df29af7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '# All ' at line 958\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '#BEGIN' at line 959\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'label ' at line 960\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'weight' at line 961\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'pose -' at line 962\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG:Nt' at line 963\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PRO_2 ' at line 964\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_3 ' at line 965\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_4 ' at line 966\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_5 ' at line 967\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_6 ' at line 968\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_7 ' at line 969\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_8 ' at line 970\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_9 ' at line 971\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_10' at line 972\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_11' at line 973\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_12' at line 974\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_13' at line 975\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_14' at line 976\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_15' at line 977\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_16' at line 978\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_17' at line 979\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_18' at line 980\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_19' at line 981\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_20' at line 982\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_21' at line 983\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_22' at line 984\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TYR_23' at line 985\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_24' at line 986\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_25' at line 987\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_26' at line 988\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_27' at line 989\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_28' at line 990\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_29' at line 991\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_30' at line 992\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_31' at line 993\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_32' at line 994\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_33' at line 995\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_34' at line 996\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_35' at line 997\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_36' at line 998\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLY_37' at line 999\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_38' at line 1000\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_39' at line 1001\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_40' at line 1002\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_41' at line 1003\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_42' at line 1004\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_43' at line 1005\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_44' at line 1006\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_45' at line 1007\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TRP_46' at line 1008\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_47' at line 1009\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_48' at line 1010\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_49' at line 1011\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_50' at line 1012\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_51' at line 1013\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_52' at line 1014\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_53' at line 1015\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE:Ct' at line 1016\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '#END_P' at line 1017\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "wt = parser.get_structure('wt', f'./tmp/ESMFoldDGEstimator_structures/wt.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5e9343d4-294e-42fc-a168-8aa3d99bb138",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "3ff3343740574401a6f065a310a0c92d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "NGLWidget()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "fb677153212c41c0882a574b63599a18",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Tab(children=(Box(children=(Box(children=(Box(children=(Label(value='step'), IntSlider(value=1, min=-100)), la…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nv.show_biopython(wt, gui=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "17867c58-98f2-40f5-90f4-5be228bec49a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '# All ' at line 1032\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '#BEGIN' at line 1033\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'label ' at line 1034\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'weight' at line 1035\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'pose -' at line 1036\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG:Nt' at line 1037\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_2 ' at line 1038\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_3 ' at line 1039\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_4 ' at line 1040\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_5 ' at line 1041\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_6 ' at line 1042\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_7 ' at line 1043\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_8 ' at line 1044\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_9 ' at line 1045\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_10' at line 1046\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_11' at line 1047\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_12' at line 1048\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_13' at line 1049\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_14' at line 1050\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_15' at line 1051\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_16' at line 1052\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_17' at line 1053\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_18' at line 1054\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_19' at line 1055\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_20' at line 1056\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_21' at line 1057\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_22' at line 1058\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_23' at line 1059\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TYR_24' at line 1060\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PRO_25' at line 1061\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_26' at line 1062\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_27' at line 1063\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_28' at line 1064\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_29' at line 1065\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_30' at line 1066\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_31' at line 1067\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_32' at line 1068\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_33' at line 1069\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_34' at line 1070\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_35' at line 1071\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_36' at line 1072\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_37' at line 1073\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLY_38' at line 1074\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_39' at line 1075\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_40' at line 1076\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_41' at line 1077\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_42' at line 1078\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_43' at line 1079\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'VAL_44' at line 1080\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_45' at line 1081\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'VAL_46' at line 1082\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TRP_47' at line 1083\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_48' at line 1084\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_49' at line 1085\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_50' at line 1086\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_51' at line 1087\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_52' at line 1088\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_53' at line 1089\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_54' at line 1090\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_55' at line 1091\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_56' at line 1092\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS:Ct' at line 1093\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '#END_P' at line 1094\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "conc = parser.get_structure('conc', f'./tmp/ESMFoldDGEstimator_structures/conc.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "824e15df-a07f-4d81-a4ff-339e54c70141",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "03ec56ec1c75450ab5dad817609da893",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "NGLWidget()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "03463825b138419ba94ef1806e3ba097",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Tab(children=(Box(children=(Box(children=(Box(children=(Label(value='step'), IntSlider(value=1, min=-100)), la…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nv.show_biopython(conc, gui=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e200b8af-132c-42e3-9034-d9b0271577ed",
   "metadata": {},
   "source": [
    "## Try with the translation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1b866c4b-43bc-4eeb-a616-3854618d25b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('../data/translations.tsv', 'r') as f:\n",
    "    lines = f.readlines()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3ee38f3b-b7a0-4741-9a33-98a5cf461815",
   "metadata": {},
   "outputs": [],
   "source": [
    "translation = lines[1].split()[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e913a3f5-c9ce-43e8-ac2f-8c07a7690c1c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 2 proteins.\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.51s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.666906 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/wt.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 54\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.203408 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/wt.pdb with minimized positions\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/trans.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/trans.pdb with minimized positions\n"
     ]
    }
   ],
   "source": [
    "estimator = ESMFoldDGEstimator(sequences=[_1enh, translation], ids=['wt', 'trans'], args=estimator_args)\n",
    "ground_results = estimator.run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0157f17a-d1fc-472c-98c1-dbd44066531b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '# All ' at line 1543\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '#BEGIN' at line 1544\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'label ' at line 1545\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'weight' at line 1546\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'pose -' at line 1547\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'MET:Nt' at line 1548\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_2 ' at line 1549\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_3 ' at line 1550\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PRO_4 ' at line 1551\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_5 ' at line 1552\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_6 ' at line 1553\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_7 ' at line 1554\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_8 ' at line 1555\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_9 ' at line 1556\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_10' at line 1557\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_11' at line 1558\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_12' at line 1559\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_13' at line 1560\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_14' at line 1561\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_15' at line 1562\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_16' at line 1563\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_17' at line 1564\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_18' at line 1565\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_19' at line 1566\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_20' at line 1567\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_21' at line 1568\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_22' at line 1569\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_23' at line 1570\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_24' at line 1571\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TYR_25' at line 1572\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_26' at line 1573\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_27' at line 1574\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_28' at line 1575\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_29' at line 1576\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_30' at line 1577\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_31' at line 1578\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_32' at line 1579\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_33' at line 1580\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_34' at line 1581\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_35' at line 1582\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_36' at line 1583\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_37' at line 1584\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_38' at line 1585\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_39' at line 1586\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_40' at line 1587\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_41' at line 1588\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_42' at line 1589\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_43' at line 1590\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_44' at line 1591\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'VAL_45' at line 1592\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_46' at line 1593\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_47' at line 1594\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TRP_48' at line 1595\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_49' at line 1596\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_50' at line 1597\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_51' at line 1598\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_52' at line 1599\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_53' at line 1600\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TYR_54' at line 1601\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_55' at line 1602\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_56' at line 1603\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_57' at line 1604\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_58' at line 1605\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_59' at line 1606\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_60' at line 1607\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_61' at line 1608\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_62' at line 1609\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_63' at line 1610\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_64' at line 1611\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_65' at line 1612\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLY_66' at line 1613\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_67' at line 1614\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_68' at line 1615\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_69' at line 1616\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_70' at line 1617\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_71' at line 1618\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'VAL_72' at line 1619\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_73' at line 1620\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_74' at line 1621\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TRP_75' at line 1622\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_76' at line 1623\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_77' at line 1624\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_78' at line 1625\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_79' at line 1626\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_80' at line 1627\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TYR_81' at line 1628\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_82' at line 1629\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_83' at line 1630\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_84' at line 1631\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_85' at line 1632\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLY:Ct' at line 1633\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '#END_P' at line 1634\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "trans_struct = parser.get_structure('trans', f'./tmp/ESMFoldDGEstimator_structures/trans.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "289e82dc-0576-467b-98cc-6dacd5b79e8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "b056337e880246b5b0759710e015c360",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "NGLWidget()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "9f44243c6ab54f909c85190b2d971026",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Tab(children=(Box(children=(Box(children=(Box(children=(Label(value='step'), IntSlider(value=1, min=-100)), la…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nv.show_biopython(trans_struct, gui=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f72fef5f-6eb3-4325-8c4f-3dbebf4d5422",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:nomelt.thermo_estimation.optimizer:Aligned sequences to determined mutation set:\n",
      "INFO:nomelt.thermo_estimation.optimizer:\n",
      "--RPRTAFSSEQLARLKREFNENRYLTERRRQQLSSELGLNEAQIKIWFQNKRAKI------------------------------\n",
      "  .|||.|..||.|||..|||..|||...||..|...|.|.|.|.||||||.|.|.                              \n",
      "MKKPRTSFTNEQIARLEKEFNRQRYLSAQRRAELAASLNLTETQVKIWFQNRRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "  Score=174\n",
      "\n",
      "INFO:nomelt.thermo_estimation.optimizer:Compressed gaps where applicable. Orignal mutation set size: 56, new size 26\n",
      "INFO:nomelt.thermo_estimation.optimizer:Aligned sequences with mutations marked:\n",
      "INFO:nomelt.thermo_estimation.optimizer:\n",
      " |  -- |  R | PRT |  A | F |  S |  S | EQ |  L | ARL |  K |  R | EFN |  E |  N \n",
      " 0     1    | === 2    | = 3    4    | == 5    | === 6    7    | === 8    9    \n",
      " |  MK |  K | PRT |  S | F |  T |  N | EQ |  I | ARL |  E |  K | EFN |  R |  Q \n",
      "\n",
      "| RYL |  T |  E |  R | RR |  Q |  Q | L |  S |  S |  E | L |  G | L |  N | E | \n",
      "| === 10   11   12   | == 13   14   | = 15   16   17   | = 18   | = 19   | = 20\n",
      "| RYL |  S |  A |  Q | RR |  A |  E | L |  A |  A |  S | L |  N | L |  T | E | \n",
      "\n",
      " A | Q |  I | KIWFQN |  K | R |  A | K |  I |  ------------------------------\n",
      "   | = 21   | ====== 22   | = 23   | = 24   25                               \n",
      " T | Q |  V | KIWFQN |  R | R |  Y | K |  T |  KRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "args = OptimizerArgs(\n",
    "    n_trials=100,\n",
    "    gapopen=-2,\n",
    "    penalize_end_gaps=False,\n",
    "    gap_compressed_mutations=True,\n",
    "    matrix='BLOSUM62',\n",
    "    sampler = optuna.samplers.NSGAIISampler(\n",
    "        population_size=10\n",
    "    )\n",
    "    \n",
    "\n",
    ")\n",
    "optimizer = MutationSubsetOptimizer(\n",
    "    _1enh, translation, \n",
    "    estimator=ESMFoldDGEstimator,\n",
    "    args=args,\n",
    "    estimator_args=estimator_args\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1766a9db-2054-4bbd-9366-3b1852b76085",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:36:56,329] A new study created in memory with name: no-name-0dcef1b8-946e-4dfc-a976-c3a7f20e084a\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f3ae96fed5e24fa284ed83fecafae83f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/100 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_9', 'mut_16', 'mut_17', 'mut_26', 'mut_31', 'mut_34', 'mut_35', 'mut_36', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFSNEQLARLEKEFNENRYLSERRRAQLAASLGLNEAQIKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n",
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:02<00:00,  2.50s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.702705 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-1643990933302938567.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.213609 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-1643990933302938567.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_12', 'mut_16', 'mut_17', 'mut_21', 'mut_22', 'mut_27', 'mut_28', 'mut_35', 'mut_38', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSSEQIARLEKEFNRQRYLTAQRRQQLSAELNLNEAQIKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:06,667] Trial 0 finished with value: 213.10730705335752 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': True, 'mut_21': False, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': False, 'mut_31': True, 'mut_32': False, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': False, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 0 with value: 213.10730705335752.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.692328 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-8422318538141388254.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.194928 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-8422318538141388254.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_17', 'mut_22', 'mut_26', 'mut_27', 'mut_31', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEKEFNEQRYLSARRRAQLSASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:11,806] Trial 1 finished with value: -183.71131741437884 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': False, 'mut_12': True, 'mut_16': True, 'mut_17': True, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': False, 'mut_32': False, 'mut_34': False, 'mut_35': True, 'mut_36': False, 'mut_38': True, 'mut_40': False, 'mut_42': False, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 1 with value: -183.71131741437884.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.672459 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/3197842581469346241.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.198964 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/3197842581469346241.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_9', 'mut_16', 'mut_17', 'mut_21', 'mut_26', 'mut_28', 'mut_34', 'mut_35', 'mut_36', 'mut_38', 'mut_42', 'mut_44', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFSNEQLARLEKEFNRNRYLSEQRRQQLAASLNLNETQVKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:17,085] Trial 2 finished with value: -202.07692975644017 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': True, 'mut_21': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_31': True, 'mut_32': False, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 2 with value: -202.07692975644017.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.13s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.761047 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-5569364559595025304.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.207635 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-5569364559595025304.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_12', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_40', 'mut_44', 'mut_53', 'mut_55']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQIARLKREFNENRYLTEQRRAELASELGLTEAQVKIWFQNKRYKT\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:21,750] Trial 3 finished with value: 704.6631934361002 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': True, 'mut_21': True, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': False, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': True, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 2 with value: -202.07692975644017.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.695412 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/514950371159373931.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue THR:CtermProteinFull 56\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.208668 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/514950371159373931.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_6', 'mut_9', 'mut_12', 'mut_16', 'mut_17', 'mut_21', 'mut_22', 'mut_26', 'mut_27', 'mut_31', 'mut_32', 'mut_34', 'mut_36', 'mut_42', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTSFSNEQIARLEKEFNRQRYLSARRRAELASSLGLNETQIKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:26,306] Trial 4 finished with value: -131.7126174584425 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': False, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': False, 'mut_38': False, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': False}. Best is trial 2 with value: -202.07692975644017.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.22s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.676222 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/3016586013763720306.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 84\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.215125 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/3016586013763720306.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_27', 'mut_35', 'mut_38', 'mut_40', 'mut_51', 'mut_53', 'mut_55']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFSNEQIARLEREFNRNRYLTARRRQQLSAELNLTEAQIKIWFQNRRYKT\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:31,322] Trial 5 finished with value: -194.30688675445705 and parameters: {'mut_0': False, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': True, 'mut_21': True, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 2 with value: -202.07692975644017.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.662278 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-1465807412345267881.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue THR:CtermProteinFull 56\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.195988 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-1465807412345267881.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_6', 'mut_9', 'mut_21', 'mut_22', 'mut_28', 'mut_32', 'mut_34', 'mut_36', 'mut_38', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTSFSNEQLARLKREFNRQRYLTEQRRQELASSLNLNEAQIKIWFQNKRAKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:35,844] Trial 6 finished with value: -122.12440556893279 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': True, 'mut_28': False, 'mut_31': False, 'mut_32': False, 'mut_34': False, 'mut_35': True, 'mut_36': False, 'mut_38': True, 'mut_40': True, 'mut_42': False, 'mut_44': False, 'mut_51': True, 'mut_53': True, 'mut_55': True, 'mut_56': False}. Best is trial 2 with value: -202.07692975644017.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.12s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.682106 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-3666001987489570057.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.206863 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-3666001987489570057.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_12', 'mut_16', 'mut_21', 'mut_22', 'mut_27', 'mut_28', 'mut_34', 'mut_35', 'mut_44', 'mut_51', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSSEQIARLEREFNRQRYLTAQRRQQLAAELGLNEAQVKIWFQNRRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:41,030] Trial 7 finished with value: -197.60238939925705 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': False, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': True, 'mut_40': False, 'mut_42': False, 'mut_44': False, 'mut_51': False, 'mut_53': False, 'mut_55': True, 'mut_56': True}. Best is trial 2 with value: -202.07692975644017.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.24s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.701117 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/5806270192307801860.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.213281 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/5806270192307801860.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_17', 'mut_21', 'mut_27', 'mut_31', 'mut_32', 'mut_34', 'mut_38', 'mut_40', 'mut_51', 'mut_53']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLKKEFNRNRYLTARRRAELASELNLTEAQIKIWFQNRRYKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:46,077] Trial 8 finished with value: -188.76831745285878 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': False, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': False, 'mut_32': False, 'mut_34': True, 'mut_35': True, 'mut_36': False, 'mut_38': False, 'mut_40': False, 'mut_42': False, 'mut_44': True, 'mut_51': True, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 2 with value: -202.07692975644017.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.36s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.713142 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/7280120194317255479.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.211649 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/7280120194317255479.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_36', 'mut_38', 'mut_40', 'mut_44', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRQELASSLNLTEAQVKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:50,808] Trial 9 finished with value: -133.50722218283144 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': False, 'mut_17': True, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': True, 'mut_28': False, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': False, 'mut_38': True, 'mut_40': True, 'mut_42': False, 'mut_44': False, 'mut_51': True, 'mut_53': True, 'mut_55': False, 'mut_56': False}. Best is trial 2 with value: -202.07692975644017.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.688469 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-6003600791278243566.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.214756 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-6003600791278243566.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_12', 'mut_16', 'mut_17', 'mut_21', 'mut_27', 'mut_28', 'mut_35', 'mut_36', 'mut_38', 'mut_51', 'mut_53', 'mut_55']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSSEQIARLEKEFNRNRYLTAQRRQQLSASLNLNEAQIKIWFQNRRYKT\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:37:56,316] Trial 10 finished with value: -203.55131822287868 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': True, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.658945 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-5353759079359576207.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue THR:CtermProteinFull 56\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.195303 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-5353759079359576207.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_6', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_22', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_42', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTSFSNEQIARLEREFNRQRYLTERRRQELAASLGLNETQIKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:00,568] Trial 11 finished with value: -124.50717708614228 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': False, 'mut_12': True, 'mut_16': True, 'mut_17': True, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': False, 'mut_32': False, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': True, 'mut_40': False, 'mut_42': False, 'mut_44': False, 'mut_51': True, 'mut_53': True, 'mut_55': True, 'mut_56': False}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.09s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.685379 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-6077842260042142050.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.213842 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-6077842260042142050.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_6', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_22', 'mut_26', 'mut_27', 'mut_32', 'mut_34', 'mut_42', 'mut_51', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTSFSNEQIARLEREFNRQRYLSARRRQELASELGLNETQIKIWFQNRRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:05,721] Trial 12 finished with value: -199.21906974681488 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.702611 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-7478381390303239529.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.215331 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-7478381390303239529.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_12', 'mut_16', 'mut_17', 'mut_21', 'mut_22', 'mut_27', 'mut_35', 'mut_40', 'mut_44', 'mut_51', 'mut_53']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQIARLEKEFNRQRYLTARRRQQLSAELGLTEAQVKIWFQNRRYKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:11,047] Trial 13 finished with value: -156.2172101375406 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': False, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': False, 'mut_51': True, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.21s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.720487 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/1539075482083187181.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.20635 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/1539075482083187181.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_8', 'mut_9', 'mut_16', 'mut_21', 'mut_22', 'mut_28', 'mut_31', 'mut_36', 'mut_42', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFTNEQLARLEREFNRQRYLTEQRRAQLSSSLGLNETQIKIWFQNKRAKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:15,492] Trial 14 finished with value: -121.8309005262677 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': True, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': False, 'mut_31': False, 'mut_32': False, 'mut_34': False, 'mut_35': True, 'mut_36': False, 'mut_38': False, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': True, 'mut_53': True, 'mut_55': False, 'mut_56': False}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.673048 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/2805021051372671701.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.20283 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/2805021051372671701.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_12', 'mut_21', 'mut_28', 'mut_34', 'mut_35', 'mut_40', 'mut_44', 'mut_51', 'mut_53']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQIARLKREFNRNRYLTEQRRQQLAAELGLTEAQVKIWFQNRRYKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:20,367] Trial 15 finished with value: 1626.0043230908163 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': False, 'mut_34': False, 'mut_35': False, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': False, 'mut_51': False, 'mut_53': False, 'mut_55': True, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.23s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.695432 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-2787655330331410616.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.196135 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-2787655330331410616.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_12', 'mut_21', 'mut_27', 'mut_28', 'mut_34', 'mut_35', 'mut_44', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSSEQIARLKREFNRNRYLTAQRRQQLAAELGLNEAQVKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:24,801] Trial 16 finished with value: -134.1899970430629 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': False, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': False, 'mut_34': True, 'mut_35': True, 'mut_36': False, 'mut_38': False, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': True, 'mut_53': True, 'mut_55': False, 'mut_56': False}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.10s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.660797 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-2956271496749707770.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.217865 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-2956271496749707770.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_17', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_34', 'mut_36', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQLARLKKEFNEQRYLSAQRRQQLASSLGLNEAQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:29,787] Trial 17 finished with value: -193.0680071550094 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': False, 'mut_12': True, 'mut_16': False, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': False, 'mut_32': False, 'mut_34': True, 'mut_35': True, 'mut_36': False, 'mut_38': False, 'mut_40': False, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.663355 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/488882420820442652.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.201568 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/488882420820442652.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_9', 'mut_12', 'mut_21', 'mut_22', 'mut_27', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_40', 'mut_44', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFSNEQIARLKREFNRQRYLTAQRRAELASELGLTEAQVKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:35,086] Trial 18 finished with value: -198.23367701881557 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': False, 'mut_17': True, 'mut_21': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_31': False, 'mut_32': False, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.12s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.705462 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-8075906530297480395.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.212592 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-8075906530297480395.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_17', 'mut_31', 'mut_35', 'mut_36', 'mut_40', 'mut_44', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEKEFNENRYLTERRRAQLSASLGLTEAQVKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:39,980] Trial 19 finished with value: -195.5830406615514 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': False, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': False, 'mut_38': False, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.11s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.67452 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-8731696028323234631.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.214251 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-8731696028323234631.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_12', 'mut_27', 'mut_28', 'mut_31', 'mut_35', 'mut_36', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQIARLKREFNENRYLTAQRRAQLSASLGLNEAQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:45,190] Trial 20 finished with value: -197.24964399222716 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': True, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_31': True, 'mut_32': False, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.687784 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-3694933774720454706.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.216898 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-3694933774720454706.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_22', 'mut_26', 'mut_27', 'mut_31', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNEQRYLSARRRAQLSASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:50,406] Trial 21 finished with value: -177.45875022510393 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': False, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': True, 'mut_32': False, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.26s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.660253 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/1240084740675316644.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.202548 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/1240084740675316644.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_17', 'mut_26', 'mut_27', 'mut_28', 'mut_31', 'mut_34', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEKEFNENRYLSAQRRAQLASSLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:38:55,663] Trial 22 finished with value: -184.86927740550743 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_31': True, 'mut_32': False, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.11s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.701853 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-7903891322781350234.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.217816 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-7903891322781350234.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_22', 'mut_32', 'mut_34', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQIARLEREFNEQRYLTERRRQELASSLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:00,715] Trial 23 finished with value: 835.0629171859485 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': True, 'mut_21': False, 'mut_22': False, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_31': True, 'mut_32': False, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.11s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.667501 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/4039050012413304332.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.204604 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/4039050012413304332.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFSNEQIARLEREFNRNRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:05,921] Trial 24 finished with value: -196.03286300588482 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 10 with value: -203.55131822287868.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.21s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.648732 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/5444693554013045497.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.200325 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/5444693554013045497.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_8', 'mut_9', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_36', 'mut_44', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFTNEQLARLKREFNEQRYLSAQRRAELASSLGLNEAQVKIWFQNKRAKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:11,302] Trial 25 finished with value: -204.5547394391137 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.08s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.699529 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/5133412783413737233.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.215952 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/5133412783413737233.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_6', 'mut_9', 'mut_16', 'mut_22', 'mut_32', 'mut_34', 'mut_35', 'mut_42', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTSFSNEQLARLEREFNEQRYLTERRRQELAAELGLNETQIKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:16,555] Trial 26 finished with value: -193.46473951716737 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': False, 'mut_17': False, 'mut_21': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': False, 'mut_55': True, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.687734 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-4976240599181666516.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.212218 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-4976240599181666516.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_6', 'mut_12', 'mut_16', 'mut_17', 'mut_21', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_32', 'mut_35', 'mut_38', 'mut_42', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTSFSSEQIARLEKEFNRQRYLSAQRRQELSAELNLNETQIKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:21,954] Trial 27 finished with value: -199.40629634261617 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': False, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.13s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.656524 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/8549671033900929541.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 84\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.195981 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/8549671033900929541.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_9', 'mut_12', 'mut_21', 'mut_22', 'mut_28', 'mut_32', 'mut_34', 'mut_36', 'mut_38', 'mut_44', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFSNEQIARLKREFNRQRYLTEQRRQELASSLNLNEAQVKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:26,898] Trial 28 finished with value: -193.17655382358114 and parameters: {'mut_0': False, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': False, 'mut_12': True, 'mut_16': True, 'mut_17': True, 'mut_21': True, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': False, 'mut_38': True, 'mut_40': False, 'mut_42': True, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.20s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.683 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-8454725583761664292.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.201193 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-8454725583761664292.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_6', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_22', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTSFSNEQIARLEREFNRQRYLTERRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:31,421] Trial 29 finished with value: 1174.2827950900912 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': False, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': True, 'mut_40': False, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.06s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.646794 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-1888582262215409744.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.206963 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-1888582262215409744.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_17', 'mut_22', 'mut_27', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEKEFNEQRYLTARRRQELAASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:36,871] Trial 30 finished with value: -203.72425688935976 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.28s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.70968 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-5941679467162108255.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.200498 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-5941679467162108255.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_28', 'mut_31', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNENRYLTEQRRAELSASLGLTEAQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:42,305] Trial 31 finished with value: -199.0359846997833 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': True, 'mut_21': False, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.11s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.714184 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-4369405698682910374.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.217769 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-4369405698682910374.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_28', 'mut_32', 'mut_34', 'mut_36', 'mut_38', 'mut_40', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFSNEQIARLEREFNRNRYLTEQRRQELASSLNLTEAQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:47,465] Trial 32 finished with value: -203.37685100181542 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.08s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.646542 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-4120471220624676067.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.204745 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-4120471220624676067.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_12', 'mut_16', 'mut_17', 'mut_22', 'mut_26', 'mut_31', 'mut_35', 'mut_36', 'mut_40', 'mut_44', 'mut_51', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQIARLEKEFNEQRYLSERRRAQLSASLGLTEAQVKIWFQNRRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:52,731] Trial 33 finished with value: -200.58223219070447 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': True, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.19s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.687029 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-3229872008045431655.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.21155 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-3229872008045431655.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_9', 'mut_16', 'mut_17', 'mut_21', 'mut_22', 'mut_27', 'mut_28', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFSNEQLARLEKEFNRQRYLTAQRRQQLSASLGLNETQVKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:39:57,814] Trial 34 finished with value: -196.765770982019 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': True, 'mut_21': False, 'mut_22': True, 'mut_26': True, 'mut_27': False, 'mut_28': False, 'mut_31': True, 'mut_32': False, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': True, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.708956 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/2753019862480926094.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.212413 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/2753019862480926094.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_12', 'mut_16', 'mut_17', 'mut_21', 'mut_22', 'mut_27', 'mut_28', 'mut_32', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQIARLEKEFNRQRYLTAQRRQELSASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:04,792] Trial 35 finished with value: -146.19774352111966 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': True, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': False, 'mut_32': False, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.12s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.680876 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-2737843721421777960.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.222254 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-2737843721421777960.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_6', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_22', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_38', 'mut_40', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTSFTNEQIARLEREFNEQRYLTEQRRQELAASLNLTEAQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:09,978] Trial 36 finished with value: -186.94258776992928 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': True, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.674101 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-5031623489995602601.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.206924 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-5031623489995602601.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_9', 'mut_21', 'mut_22', 'mut_28', 'mut_32', 'mut_34', 'mut_36', 'mut_38', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFSNEQLARLKREFNRQRYLTEQRRQELASSLNLNEAQIKIWFQNKRAKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:15,604] Trial 37 finished with value: -201.02254316073956 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': True, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.80s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.69698 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-8654930111492822419.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.206664 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-8654930111492822419.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_17', 'mut_22', 'mut_26', 'mut_27', 'mut_28', 'mut_34', 'mut_36', 'mut_44', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQLARLKKEFNEQRYLSAQRRQQLASSLGLNEAQVKIWFQNKRAKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:21,698] Trial 38 finished with value: -197.5455610674269 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': False, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': True, 'mut_40': False, 'mut_42': False, 'mut_44': False, 'mut_51': False, 'mut_53': False, 'mut_55': True, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.07s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.656695 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/8880879142576014734.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.21656 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/8880879142576014734.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFSNEQLARLEREFNRNRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:26,976] Trial 39 finished with value: -197.18580959510692 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': False, 'mut_17': True, 'mut_21': False, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': True, 'mut_31': False, 'mut_32': False, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': False, 'mut_55': False, 'mut_56': True}. Best is trial 25 with value: -204.5547394391137.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.20s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.675337 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/7593271020313224592.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.22728 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/7593271020313224592.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFSNEQIARLEREFNRNRYLTERRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:31,993] Trial 40 finished with value: -204.5769848686726 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 40 with value: -204.5769848686726.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.14s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.678623 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/8921233843265656351.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.20799 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/8921233843265656351.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_16', 'mut_22', 'mut_31', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQLARLEREFNEQRYLTERRRAELSASLGLTEAQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:37,299] Trial 41 finished with value: -196.7804046037423 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 40 with value: -204.5769848686726.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.65186 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-1366271193971457752.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.207691 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-1366271193971457752.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_6', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTSFSNEQIARLEREFNRNRYLTERRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:42,492] Trial 42 finished with value: -192.63296243892722 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_31': True, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 40 with value: -204.5769848686726.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.22s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.67834 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-1427335291782847103.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.196933 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-1427335291782847103.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:47,692] Trial 43 finished with value: -200.04634975684658 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 40 with value: -204.5769848686726.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.15s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.687348 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.216074 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_31', 'mut_32', 'mut_35', 'mut_36', 'mut_38', 'mut_40', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRAELSASLNLTEAQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:52,895] Trial 44 finished with value: -206.92404440092264 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 44 with value: -206.92404440092264.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.07s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.719815 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-934833048698934043.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.210685 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-934833048698934043.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_17', 'mut_21', 'mut_26', 'mut_27', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEKEFNRNRYLSARRRQELSASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:40:58,037] Trial 45 finished with value: -197.19102582409332 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': True, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 44 with value: -206.92404440092264.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.21s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.71775 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/7284633217467068017.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.211017 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/7284633217467068017.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_16', 'mut_26', 'mut_28', 'mut_31', 'mut_32', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQLARLEREFNENRYLSEQRRAELSASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:03,124] Trial 46 finished with value: -197.47919672952986 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': True, 'mut_21': True, 'mut_22': False, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 44 with value: -206.92404440092264.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.698106 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-6688902047467748142.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.206081 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-6688902047467748142.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_22', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFSNEQIARLEREFNRQRYLTEQRRQELAASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:08,458] Trial 47 finished with value: -207.7674090227535 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.08s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.699112 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-2409970090031215444.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.211323 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-2409970090031215444.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_17', 'mut_22', 'mut_27', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_42', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEKEFNEQRYLTARRRQELAASLGLNETQIKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:13,877] Trial 48 finished with value: -198.67198599429494 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.10s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.662095 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/112442811043290879.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.196858 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/112442811043290879.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNRNRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:19,121] Trial 49 finished with value: -115.35390958488827 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': True, 'mut_21': False, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.27s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.689295 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.199211 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTSEQIARLEREFNENRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:24,062] Trial 50 finished with value: -206.459219630012 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.05s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.664443 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-2082251672451374761.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.207234 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-2082251672451374761.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:28,963] Trial 51 finished with value: -204.37019986698445 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': False, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.22s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.657087 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.199478 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNRNRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:34,148] Trial 52 finished with value: -206.92404440092264 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.09s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.678198 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.208449 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_38', 'mut_40', 'mut_44', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFTNEQIARLEREFNRNRYLTEQRRQELAASLNLTEAQVKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:39,108] Trial 53 finished with value: -206.459219630012 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.11s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.638467 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-1129618074483644476.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.200207 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-1129618074483644476.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNENRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:43,841] Trial 54 finished with value: -199.24393268810232 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': True, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.695939 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/5943283024854124266.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.208154 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/5943283024854124266.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_26', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNENRYLSEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:48,883] Trial 55 finished with value: -205.5700255321133 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.08s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.646554 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/790380330174303460.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.213763 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/790380330174303460.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRQELSASLGLTEAQIKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:53,766] Trial 56 finished with value: -206.25816985313068 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.20s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.698628 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/2680116683633943556.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.212806 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/2680116683633943556.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_6', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTSFSNEQIARLEREFNRNRYLTERRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:41:58,816] Trial 57 finished with value: -194.7769426464949 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': False, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.657141 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-1427335291782847103.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.201606 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-1427335291782847103.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQIARLEREFNRNRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:03,823] Trial 58 finished with value: -200.04634975684658 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.13s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.702814 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-2245311687107380586.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.204881 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-2245311687107380586.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQIARLEREFNENRYLTEQRRQELASSLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:08,949] Trial 59 finished with value: -203.1715103861931 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.10s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.679931 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/8017059900954798065.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.201293 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/8017059900954798065.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQIARLEREFNRNRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:13,531] Trial 60 finished with value: 309.3116137156714 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.24s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.677478 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-9016285338723023963.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.204 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-9016285338723023963.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_16', 'mut_28', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQLARLEREFNENRYLTEQRRQELSASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:18,591] Trial 61 finished with value: -199.9832562526361 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.14s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.680127 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-8717831501988336815.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.215908 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-8717831501988336815.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_55', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNRNRYLTEQRRQELAASLGLTETQVKIWFQNKRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:23,824] Trial 62 finished with value: -198.60481854981083 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.662261 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-2236627439574813908.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.193822 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-2236627439574813908.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_28', 'mut_31', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNENRYLTEQRRAELSASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:28,752] Trial 63 finished with value: -202.74581961466583 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': True, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.09s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.712591 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-6799172454207078505.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.212516 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-6799172454207078505.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNENRYLTEQRRQELAASLGLTETQIKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:33,830] Trial 64 finished with value: -197.42152483553633 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.08s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.645088 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-727171274254451742.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.208573 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-727171274254451742.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNRNRYLTEQRRQELAASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:38,809] Trial 65 finished with value: -185.48682460566206 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.684505 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/2852433320761535327.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.205696 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/2852433320761535327.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNRNRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:43,860] Trial 66 finished with value: -187.5616291195559 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.06s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.641304 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.197891 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_26', 'mut_28', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLSEQRRQELSASLGLTETQVKIWFQNKRYKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:48,493] Trial 67 finished with value: -206.459219630012 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.08s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.692885 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/3485802508966683917.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.207872 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/3485802508966683917.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQLARLEREFNENRYLTEQRRQELAASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:52,907] Trial 68 finished with value: -130.32332343884252 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': False}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.09s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.633731 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-2052261638460595579.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.206444 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-2052261638460595579.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_16', 'mut_21', 'mut_26', 'mut_28', 'mut_31', 'mut_32', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQLARLEREFNRNRYLSEQRRAELSASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:42:57,563] Trial 69 finished with value: 367.7009311987932 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.20s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.692843 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-5926174174207126010.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.211604 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-5926174174207126010.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNENRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:02,533] Trial 70 finished with value: 367.8783472002406 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.10s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.640724 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-6630870044849721105.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.202244 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-6630870044849721105.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_21', 'mut_27', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNRNRYLTAQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:07,337] Trial 71 finished with value: -203.99283647386534 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.22s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.679673 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-8362378010902189759.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.209573 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-8362378010902189759.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNENRYLTEQRRAELAASLGLTETQIKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:12,328] Trial 72 finished with value: -204.96573358401182 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.14s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.72037 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/6208077350913173518.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.212295 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/6208077350913173518.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:17,640] Trial 73 finished with value: -190.93276511042768 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.56s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.717986 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-25125053887958949.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.208906 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-25125053887958949.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFSNEQLARLEREFNRNRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:22,791] Trial 74 finished with value: -196.88193854487207 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.19s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.693872 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-1427464316970288232.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.216437 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-1427464316970288232.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_26', 'mut_28', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQIARLEREFNRNRYLSEQRRQELSASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:27,875] Trial 75 finished with value: -201.01327376735028 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.19s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.657211 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/2145758531601543305.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.197938 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/2145758531601543305.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:32,737] Trial 76 finished with value: -193.8005496536101 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.690598 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.214208 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNRNRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:38,104] Trial 77 finished with value: -206.92404440092264 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.13s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.689044 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.206717 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_21', 'mut_22', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNRQRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:43,054] Trial 78 finished with value: -206.459219630012 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.696723 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-1606716370206802563.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.205221 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-1606716370206802563.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_21', 'mut_26', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNRNRYLSEQRRAELAASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:48,364] Trial 79 finished with value: -178.72683786343129 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.11s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.668559 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/3362272095551928277.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.211037 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/3362272095551928277.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRQELAASLGLNEAQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:53,893] Trial 80 finished with value: -188.59862546405057 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.12s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.672501 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/9030403596217771644.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.214718 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/9030403596217771644.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQIARLEREFNENRYLTEQRRQELSASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:43:59,335] Trial 81 finished with value: -201.1351200174784 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.669284 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-8668677323029124376.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.208455 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-8668677323029124376.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNRNRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:04,517] Trial 82 finished with value: -198.78162329062198 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.19s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.711559 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/2528509473433527516.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.211747 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/2528509473433527516.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNRNRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:09,728] Trial 83 finished with value: -205.99240653041997 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.21s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.64935 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/8114836266568431090.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.211557 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/8114836266568431090.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_28', 'mut_31', 'mut_32', 'mut_35', 'mut_36', 'mut_38', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNENRYLTEQRRAELSASLNLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:14,620] Trial 84 finished with value: 887.6889653654409 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.722475 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/4525099730535130334.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.219357 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/4525099730535130334.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNRNRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:19,909] Trial 85 finished with value: -199.6269208509751 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': True, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.11s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.656586 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.204733 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_26', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLSEQRRAELAASLGLTETQVKIWFQNKRYKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:24,867] Trial 86 finished with value: -206.459219630012 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.23s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.689639 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-8618400060478764145.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.209142 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-8618400060478764145.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:29,481] Trial 87 finished with value: -138.271620658055 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': False}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.25s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.697744 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.202609 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_38', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRQELAASLNLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:34,897] Trial 88 finished with value: -206.92404440092264 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.663098 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/6670537053437218969.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.197683 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/6670537053437218969.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_16', 'mut_26', 'mut_28', 'mut_31', 'mut_32', 'mut_35', 'mut_36', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQLARLEREFNENRYLSEQRRAELSASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:40,367] Trial 89 finished with value: -197.24926039224533 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': True, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.663233 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-6688902047467748142.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.203677 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-6688902047467748142.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNRNRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:45,695] Trial 90 finished with value: -207.7674090227535 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': False, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.11s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.701713 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.201553 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-3393609014009485835.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:50,582] Trial 91 finished with value: -206.459219630012 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.20s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.708423 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.211053 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_16', 'mut_26', 'mut_28', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTSFTNEQLARLEREFNENRYLSEQRRQELSASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:44:55,953] Trial 92 finished with value: -206.92404440092264 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.15s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.688774 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-4513386468334631745.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.211432 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-4513386468334631745.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLEREFNRNRYLTEQRRAELSASLGLTETQVKIWFQNKRYKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:45:01,130] Trial 93 finished with value: -203.48057536920197 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.15s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.692476 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/-144460049307651124.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.207689 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/-144460049307651124.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_21', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNRNRYLTEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:45:05,671] Trial 94 finished with value: -129.88472977860897 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': False, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': False}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.23s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.697663 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/2485736975971610887.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.196597 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/2485736975971610887.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQIARLEREFNENRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:45:10,534] Trial 95 finished with value: -205.31557123251815 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.17s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.698359 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.216795 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/1720276124906871372.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_26', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNENRYLSEQRRAELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:45:15,872] Trial 96 finished with value: -206.92404440092264 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.21s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.689577 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/790380330174303460.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.213689 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/790380330174303460.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_9', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFSNEQLARLKREFNENRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:45:20,974] Trial 97 finished with value: -206.25816985313068 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': True, 'mut_31': True, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.16s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.688538 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/8443835617907475865.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.213408 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/8443835617907475865.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_16', 'mut_21', 'mut_28', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_40', 'mut_42', 'mut_44', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLEREFNRNRYLTEQRRQELAASLGLTETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:45:26,643] Trial 98 finished with value: -187.51065308761827 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': False, 'mut_17': False, 'mut_21': False, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                     | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:01<00:00,  1.18s/it]\u001b[A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.68281 seconds.\n",
      "core.import_pose.import_pose: File './tmp/ESMFoldDGEstimator_structures/1645899727866714080.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue GLY:CtermProteinFull 86\n",
      "core.scoring.etable: Starting energy table calculation\n",
      "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "core.scoring.etable: Finished calculating energy tables.\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.209613 seconds to load from binary\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file ./tmp/ESMFoldDGEstimator_structures/1645899727866714080.pdb with minimized positions\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-27 15:45:31,698] Trial 99 finished with value: -202.10521979763718 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': False, 'mut_40': True, 'mut_42': True, 'mut_44': True, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 47 with value: -207.7674090227535.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "({'mut_0': True,\n",
       "  'mut_2': True,\n",
       "  'mut_6': True,\n",
       "  'mut_8': True,\n",
       "  'mut_9': True,\n",
       "  'mut_12': False,\n",
       "  'mut_16': True,\n",
       "  'mut_17': False,\n",
       "  'mut_21': False,\n",
       "  'mut_22': False,\n",
       "  'mut_26': True,\n",
       "  'mut_27': False,\n",
       "  'mut_28': True,\n",
       "  'mut_31': True,\n",
       "  'mut_32': True,\n",
       "  'mut_34': False,\n",
       "  'mut_35': True,\n",
       "  'mut_36': True,\n",
       "  'mut_38': False,\n",
       "  'mut_40': False,\n",
       "  'mut_42': True,\n",
       "  'mut_44': True,\n",
       "  'mut_51': False,\n",
       "  'mut_53': True,\n",
       "  'mut_55': False,\n",
       "  'mut_56': True},\n",
       " -207.7674090227535,\n",
       " <optuna.study.study.Study at 0x147199374e80>)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "optimizer.run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "5a0346d1-395f-42b7-87ce-453c7564f3a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MKKPRTSFTNEQLARLEREFNENRYLSEQRRAELSASLGLNETQVKIWFQNKRYKIKRKQINQLLGLTEQQVKIWFQNRRYKTKRG'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "optimizer.optimized_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7b42c4b6-cec8-42cb-a09b-22e1620d6ff0",
   "metadata": {},
   "outputs": [],
   "source": [
    "best_file_hash = optimizer.hash_mutation_set(optimizer.best_mutations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f84db9d6-054e-4eab-82f9-dc1705ae5f3a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-6688902047467748142"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "best_file_hash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "5ff40889-b397-4907-81a5-2a9e327d5644",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '# All ' at line 1535\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '#BEGIN' at line 1536\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'label ' at line 1537\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'weight' at line 1538\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'pose -' at line 1539\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'MET:Nt' at line 1540\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_2 ' at line 1541\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_3 ' at line 1542\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PRO_4 ' at line 1543\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_5 ' at line 1544\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_6 ' at line 1545\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_7 ' at line 1546\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_8 ' at line 1547\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_9 ' at line 1548\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_10' at line 1549\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_11' at line 1550\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_12' at line 1551\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_13' at line 1552\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_14' at line 1553\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_15' at line 1554\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_16' at line 1555\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_17' at line 1556\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_18' at line 1557\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_19' at line 1558\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_20' at line 1559\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_21' at line 1560\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_22' at line 1561\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_23' at line 1562\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_24' at line 1563\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TYR_25' at line 1564\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_26' at line 1565\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_27' at line 1566\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_28' at line 1567\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_29' at line 1568\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_30' at line 1569\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_31' at line 1570\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_32' at line 1571\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_33' at line 1572\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_34' at line 1573\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_35' at line 1574\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ALA_36' at line 1575\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'SER_37' at line 1576\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_38' at line 1577\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLY_39' at line 1578\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_40' at line 1579\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_41' at line 1580\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_42' at line 1581\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_43' at line 1582\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_44' at line 1583\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'VAL_45' at line 1584\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_46' at line 1585\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_47' at line 1586\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TRP_48' at line 1587\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_49' at line 1588\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_50' at line 1589\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_51' at line 1590\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_52' at line 1591\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_53' at line 1592\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TYR_54' at line 1593\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_55' at line 1594\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_56' at line 1595\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_57' at line 1596\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_58' at line 1597\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_59' at line 1598\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_60' at line 1599\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_61' at line 1600\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_62' at line 1601\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_63' at line 1602\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_64' at line 1603\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_65' at line 1604\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLY_66' at line 1605\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LEU_67' at line 1606\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_68' at line 1607\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLU_69' at line 1608\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_70' at line 1609\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_71' at line 1610\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'VAL_72' at line 1611\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_73' at line 1612\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ILE_74' at line 1613\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TRP_75' at line 1614\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'PHE_76' at line 1615\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLN_77' at line 1616\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ASN_78' at line 1617\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_79' at line 1618\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_80' at line 1619\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'TYR_81' at line 1620\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_82' at line 1621\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'THR_83' at line 1622\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'LYS_84' at line 1623\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'ARG_85' at line 1624\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record 'GLY:Ct' at line 1625\n",
      "  warnings.warn(\n",
      "/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/Bio/PDB/PDBParser.py:395: PDBConstructionWarning: Ignoring unrecognized record '#END_P' at line 1626\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "struct = parser.get_structure('opt', f'./tmp/ESMFoldDGEstimator_structures/{best_file_hash}.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b9b9650b-abe2-40e9-bb39-2652fdf11c03",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "8dc6a6e6ce9b40f5962abd7c3725c5c3",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "NGLWidget()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "ea281a893daf45a9b7f819129ca74a75",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Tab(children=(Box(children=(Box(children=(Box(children=(Label(value='step'), IntSlider(value=1, min=-100)), la…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nv.show_biopython(struct, gui=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83e1dab3-773d-465d-ae33-181785b699b2",
   "metadata": {},
   "source": [
    "### repeat with cut version of translation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3b788ebf-7d84-4f1b-b317-12bddfe11b5c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:nomelt.thermo_estimation.optimizer:Aligned sequences to determined mutation set:\n",
      "INFO:nomelt.thermo_estimation.optimizer:\n",
      "--RPRTAFSSEQLARLKREFNENRYLTERRRQQLSSELGLNEAQIKIWFQNKRAKI------------------------------\n",
      "  .|||.|..||.|||..|||..|||...||..|...|.|.|.|.||||||.|.|.                              \n",
      "MKKPRTSFTNEQIARLEKEFNRQRYLSAQRRAELAASLNLTETQVKIWFQNRRYKTKRKQINQLLGLTEQQVKIWFQNRRYKTKRG\n",
      "  Score=174\n",
      "\n",
      "INFO:nomelt.thermo_estimation.optimizer:Aligned sequences after cutting tails:\n",
      "\n",
      "INFO:nomelt.thermo_estimation.optimizer:--RPRTAFSSEQLARLKREFNENRYLTERRRQQLSSELGLNEAQIKIWFQNKRAKI--\n",
      "  .|||.|..||.|||..|||..|||...||..|...|.|.|.|.||||||.|.|.  \n",
      "MKKPRTSFTNEQIARLEKEFNRQRYLSAQRRAELAASLNLTETQVKIWFQNRRYKTKR\n",
      "INFO:nomelt.thermo_estimation.optimizer:Compressed gaps where applicable. Orignal mutation set size: 28, new size 26\n",
      "INFO:nomelt.thermo_estimation.optimizer:Aligned sequences with mutations marked:\n",
      "INFO:nomelt.thermo_estimation.optimizer:\n",
      " |  -- |  R | PRT |  A | F |  S |  S | EQ |  L | ARL |  K |  R | EFN |  E |  N \n",
      " 0     1    | === 2    | = 3    4    | == 5    | === 6    7    | === 8    9    \n",
      " |  MK |  K | PRT |  S | F |  T |  N | EQ |  I | ARL |  E |  K | EFN |  R |  Q \n",
      "\n",
      "| RYL |  T |  E |  R | RR |  Q |  Q | L |  S |  S |  E | L |  G | L |  N | E | \n",
      "| === 10   11   12   | == 13   14   | = 15   16   17   | = 18   | = 19   | = 20\n",
      "| RYL |  S |  A |  Q | RR |  A |  E | L |  A |  A |  S | L |  N | L |  T | E | \n",
      "\n",
      " A | Q |  I | KIWFQN |  K | R |  A | K |  I |  --\n",
      "   | = 21   | ====== 22   | = 23   | = 24   25   \n",
      " T | Q |  V | KIWFQN |  R | R |  Y | K |  T |  KR\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "args = OptimizerArgs(\n",
    "    n_trials=100,\n",
    "    gapopen=-2,\n",
    "    penalize_end_gaps=False,\n",
    "    gap_compressed_mutations=True,\n",
    "    matrix='BLOSUM62',\n",
    "    sampler = optuna.samplers.NSGAIISampler(\n",
    "        population_size=10\n",
    "    ),\n",
    "    cut_tails=2\n",
    "    \n",
    "\n",
    ")\n",
    "optimizer = MutationSubsetOptimizer(\n",
    "    _1enh, translation, \n",
    "    estimator=ESMFoldDGEstimator,\n",
    "    args=args,\n",
    "    estimator_args=estimator_args\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "156cb1f0-1739-4696-bff2-ef46c09f7da5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[I 2023-07-31 15:42:10,844] A new study created in memory with name: mutation_optimizer_run\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2b5cc531a5a7453ca464f422809231ba",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/100 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_6', 'mut_8', 'mut_21', 'mut_28', 'mut_34', 'mut_36', 'mut_38', 'mut_42']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTSFTSEQLARLKREFNRNRYLTEQRRQQLASSLNLNETQIKIWFQNKRAKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n",
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:02<00:00,  2.19s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1572274202 seed_offset=0 real_seed=1572274202\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1572274202 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.678021 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5784497870654574083.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 54\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.215958 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5784497870654574083.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_2', 'mut_8', 'mut_9', 'mut_17', 'mut_21', 'mut_22', 'mut_26', 'mut_27', 'mut_36', 'mut_38', 'mut_40', 'mut_44']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKKPRTAFTNEQLARLKKEFNRQRYLSARRRQQLSSSLNLTEAQVKIWFQNKRAKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-31 15:43:24,863] Trial 0 finished with value: -134.88408045718248 and parameters: {'mut_0': False, 'mut_2': False, 'mut_6': True, 'mut_8': True, 'mut_9': False, 'mut_12': False, 'mut_16': False, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': False, 'mut_27': False, 'mut_28': True, 'mut_31': False, 'mut_32': False, 'mut_34': True, 'mut_35': False, 'mut_36': True, 'mut_38': True, 'mut_40': False, 'mut_42': True, 'mut_44': False, 'mut_51': False, 'mut_53': False, 'mut_55': False, 'mut_56': False}. Best is trial 0 with value: -134.88408045718248.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.06s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1413699497 seed_offset=0 real_seed=1413699497\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1413699497 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.640061 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2116316652191260170.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ILE:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.204393 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2116316652191260170.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_2', 'mut_6', 'mut_8', 'mut_9', 'mut_12', 'mut_16', 'mut_22', 'mut_31', 'mut_32', 'mut_38', 'mut_40', 'mut_42', 'mut_51', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: KPRTSFTNEQIARLEREFNEQRYLTERRRAELSSELNLTETQIKIWFQNRRYKIKR\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-31 15:43:29,420] Trial 1 finished with value: -123.59349979085002 and parameters: {'mut_0': True, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': True, 'mut_12': False, 'mut_16': False, 'mut_17': True, 'mut_21': True, 'mut_22': True, 'mut_26': True, 'mut_27': True, 'mut_28': False, 'mut_31': False, 'mut_32': False, 'mut_34': False, 'mut_35': False, 'mut_36': True, 'mut_38': True, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': False, 'mut_53': False, 'mut_55': False, 'mut_56': False}. Best is trial 0 with value: -134.88408045718248.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.06s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-725379855 seed_offset=0 real_seed=-725379855\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-725379855 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.639565 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3612855936940394784.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ARG:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.193191 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3612855936940394784.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_9', 'mut_21', 'mut_22', 'mut_27', 'mut_28', 'mut_31', 'mut_34', 'mut_35', 'mut_36', 'mut_38', 'mut_44', 'mut_51', 'mut_53', 'mut_55']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTAFSNEQLARLKREFNRQRYLTAQRRAQLAASLNLNEAQVKIWFQNRRYKT\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-31 15:43:33,804] Trial 2 finished with value: -133.9377455366315 and parameters: {'mut_0': False, 'mut_2': True, 'mut_6': True, 'mut_8': True, 'mut_9': True, 'mut_12': True, 'mut_16': True, 'mut_17': False, 'mut_21': False, 'mut_22': True, 'mut_26': False, 'mut_27': False, 'mut_28': False, 'mut_31': True, 'mut_32': True, 'mut_34': False, 'mut_35': False, 'mut_36': False, 'mut_38': True, 'mut_40': True, 'mut_42': True, 'mut_44': False, 'mut_51': True, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 0 with value: -134.88408045718248.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.06s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1658783531 seed_offset=0 real_seed=1658783531\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1658783531 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.641795 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8745981522794528134.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue THR:CtermProteinFull 54\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.211491 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8745981522794528134.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_6', 'mut_9', 'mut_16', 'mut_21', 'mut_22', 'mut_27', 'mut_28', 'mut_31', 'mut_34', 'mut_38', 'mut_40', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTSFSNEQLARLEREFNRQRYLTAQRRAQLASELNLTEAQIKIWFQNKRYKIKR\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-31 15:43:38,143] Trial 3 finished with value: -131.19105693573246 and parameters: {'mut_0': False, 'mut_2': False, 'mut_6': False, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': False, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': True, 'mut_32': False, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': True, 'mut_40': False, 'mut_42': False, 'mut_44': True, 'mut_51': True, 'mut_53': True, 'mut_55': True, 'mut_56': False}. Best is trial 0 with value: -134.88408045718248.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.06s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-470385722 seed_offset=0 real_seed=-470385722\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-470385722 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.639858 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2886536536570925468.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ARG:CtermProteinFull 58\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.202303 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2886536536570925468.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_2', 'mut_8', 'mut_12', 'mut_21', 'mut_26', 'mut_32', 'mut_34', 'mut_35', 'mut_36', 'mut_38', 'mut_40', 'mut_42', 'mut_51', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: KPRTAFTSEQIARLKREFNRNRYLSERRRQELAASLNLTETQIKIWFQNRRYKIKR\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-31 15:43:42,333] Trial 4 finished with value: -138.35784833789904 and parameters: {'mut_0': True, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': False, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': True, 'mut_32': False, 'mut_34': True, 'mut_35': False, 'mut_36': False, 'mut_38': True, 'mut_40': True, 'mut_42': False, 'mut_44': False, 'mut_51': False, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 4 with value: -138.35784833789904.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.05s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=1910722689 seed_offset=0 real_seed=1910722689\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=1910722689 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.640128 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7715858856613547404.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ARG:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.20313 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7715858856613547404.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_6', 'mut_9', 'mut_16', 'mut_17', 'mut_21', 'mut_22', 'mut_27', 'mut_28', 'mut_32', 'mut_36', 'mut_38', 'mut_40', 'mut_44', 'mut_51', 'mut_53', 'mut_56']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: RPRTSFSNEQLARLEKEFNRQRYLTAQRRQELSSSLNLTEAQVKIWFQNRRYKIKR\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-31 15:43:46,681] Trial 5 finished with value: -139.21237435239811 and parameters: {'mut_0': False, 'mut_2': True, 'mut_6': False, 'mut_8': True, 'mut_9': False, 'mut_12': True, 'mut_16': False, 'mut_17': False, 'mut_21': True, 'mut_22': False, 'mut_26': True, 'mut_27': False, 'mut_28': False, 'mut_31': False, 'mut_32': True, 'mut_34': True, 'mut_35': True, 'mut_36': True, 'mut_38': True, 'mut_40': True, 'mut_42': True, 'mut_44': False, 'mut_51': True, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 5 with value: -139.21237435239811.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A\n",
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.13s/it]\u001b[A\n",
      "INFO:pyrosetta.rosetta:Found rosetta database at: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database; using it....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyrosetta.rosetta:PyRosetta-4 2023 [Rosetta PyRosetta4.Release.python39.ubuntu 2023.27+release.e3ce6ea9faf661ae8fa769511e2a9b8596417e58 2023-07-07T12:00:46] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "INFO:rosetta:core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
      "INFO:rosetta:core.init: Rosetta version: PyRosetta4.Release.python39.ubuntu r353 2023.27+release.e3ce6ea9faf e3ce6ea9faf661ae8fa769511e2a9b8596417e58 http://www.pyrosetta.org 2023-07-07T12:00:46\n",
      "INFO:rosetta:core.init: command: PyRosetta -ex1 -ex2aro -database /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database\n",
      "INFO:rosetta:basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=670257654 seed_offset=0 real_seed=670257654\n",
      "INFO:rosetta:basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=670257654 RG_type=mt19937\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
      "INFO:rosetta:core.chemical.GlobalResidueTypeSet: Total time to initialize 0.642659 seconds.\n",
      "INFO:rosetta:core.import_pose.import_pose: File '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6322234547685451747.pdb' automatically determined to be of type PDB\n",
      "INFO:rosetta:core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ARG:CtermProteinFull 56\n",
      "INFO:rosetta:core.scoring.etable: Starting energy table calculation\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
      "INFO:rosetta:core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
      "INFO:rosetta:core.scoring.etable: Finished calculating energy tables.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
      "INFO:rosetta:core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
      "INFO:rosetta:basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
      "INFO:rosetta:core.scoring.elec.util: Read 40 countpair representative atoms\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/mmfs1/gscratch/cheme/usr/evankomp/mamba/mambaforge/envs/nomelt2/lib/python3.9/site-packages/pyrosetta-2023.27+release.e3ce6ea9faf-py3.9-linux-x86_64.egg/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "INFO:rosetta:core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.210879 seconds to load from binary\n",
      "DEBUG:nomelt.thermo_estimation.rosetta:Updated PDB file /mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6322234547685451747.pdb with minimized positions\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Calling estimator with mutation subset: ['mut_0', 'mut_8', 'mut_12', 'mut_17', 'mut_21', 'mut_26', 'mut_28', 'mut_31', 'mut_32', 'mut_34', 'mut_35', 'mut_38', 'mut_40', 'mut_51', 'mut_53']\n",
      "DEBUG:nomelt.thermo_estimation.optimizer:Variant sequence: MKRPRTAFTSEQIARLKKEFNRNRYLSEQRRAELAAELNLTEAQIKIWFQNRRYKI\n",
      "INFO:nomelt.thermo_estimation.estimators:Running ESMfold on 1 proteins.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-07-31 15:43:52,762] Trial 6 finished with value: -128.88096008537264 and parameters: {'mut_0': False, 'mut_2': False, 'mut_6': True, 'mut_8': False, 'mut_9': True, 'mut_12': False, 'mut_16': True, 'mut_17': True, 'mut_21': True, 'mut_22': True, 'mut_26': False, 'mut_27': True, 'mut_28': True, 'mut_31': False, 'mut_32': True, 'mut_34': False, 'mut_35': False, 'mut_36': True, 'mut_38': True, 'mut_40': True, 'mut_42': False, 'mut_44': True, 'mut_51': True, 'mut_53': True, 'mut_55': False, 'mut_56': True}. Best is trial 5 with value: -139.21237435239811.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                                                                                                                             | 0/1 [00:00<?, ?it/s]\u001b[A"
     ]
    }
   ],
   "source": [
    "optimizer.run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b7d4f345-9cd7-440a-a72d-df673ffeb3ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'KPRTAFSSEQLARLEKEFNRNRYLSERRRQELAASLNLTEAQVKIWFQNKRYKIKR'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "optimizer.optimized_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4745452e-0ba8-47ca-8c2b-e1583262abe3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(0,\n",
       "  'MKRPRTSFTSEQLARLEKEFNRQRYLTARRRAQLSSELNLTEAQVKIWFQNKRAKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7305603416206835138.pdb'),\n",
       " (1,\n",
       "  'MKKPRTAFSNEQIARLKKEFNRNRYLSEQRRQELSSELGLTEAQVKIWFQNRRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-8907532345303521824.pdb'),\n",
       " (2,\n",
       "  'KPRTSFSSEQLARLKKEFNEQRYLSAQRRAELASSLNLTETQIKIWFQNRRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6370904440427634447.pdb'),\n",
       " (3,\n",
       "  'MKKPRTSFTNEQIARLEKEFNRNRYLTAQRRQQLSSSLNLNEAQIKIWFQNKRAKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3474360579718080596.pdb'),\n",
       " (4,\n",
       "  'MKKPRTAFTNEQIARLKREFNRQRYLTARRRQELSSELGLTEAQVKIWFQNRRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-8776918149564386177.pdb'),\n",
       " (5,\n",
       "  'KPRTAFTNEQLARLEKEFNEQRYLSAQRRQQLAAELNLNEAQIKIWFQNRRAKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5977496160613348706.pdb'),\n",
       " (6,\n",
       "  'MKRPRTSFTSEQIARLEKEFNENRYLSERRRAQLSSELGLTETQIKIWFQNKRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2456568645132258140.pdb'),\n",
       " (7,\n",
       "  'MKKPRTAFTSEQLARLKKEFNRNRYLSARRRAQLAASLGLNETQVKIWFQNRRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/4732908302256287683.pdb'),\n",
       " (8,\n",
       "  'MKKPRTAFTNEQIARLEKEFNRQRYLSAQRRAQLSSSLGLTEAQVKIWFQNRRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1687671349997668933.pdb'),\n",
       " (9,\n",
       "  'KPRTSFSSEQLARLEREFNRQRYLTARRRQELASSLNLTETQIKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5978213321369291198.pdb'),\n",
       " (10,\n",
       "  'KPRTSFTNEQLARLEKEFNEQRYLSAQRRQQLSAELNLNEAQVKIWFQNRRAKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7616826071661193478.pdb'),\n",
       " (11,\n",
       "  'KPRTSFTSEQIARLEKEFNRQRYLTAQRRQELSSSLNLTEAQIKIWFQNKRAKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6596966036710300651.pdb'),\n",
       " (12,\n",
       "  'KPRTAFTNEQLARLEREFNEQRYLSARRRQELAAELNLTEAQIKIWFQNKRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1559454905206351569.pdb'),\n",
       " (13,\n",
       "  'KPRTAFTNEQIARLEKEFNENRYLSARRRQQLSSELGLTETQIKIWFQNKRAKT',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4444338000866611062.pdb'),\n",
       " (14,\n",
       "  'KPRTAFTSEQIARLKKEFNRNRYLSEQRRAELAASLGLNETQVKIWFQNRRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/4023925123066691254.pdb'),\n",
       " (15,\n",
       "  'MKRPRTSFSSEQIARLEKEFNENRYLSEQRRAQLSSSLGLTETQIKIWFQNKRYKT',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3346342667759875874.pdb'),\n",
       " (16,\n",
       "  'MKRPRTSFTSEQIARLEKEFNRQRYLTARRRAQLSASLNLTEAQIKIWFQNKRAKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/4108140838209054598.pdb'),\n",
       " (17,\n",
       "  'RPRTAFTSEQLARLEKEFNEQRYLSARRRQQLASELNLTEAQIKIWFQNRRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/7472929288416652578.pdb'),\n",
       " (18,\n",
       "  'KPRTSFSSEQLARLKREFNRNRYLSARRRQQLAASLNLNETQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6066038780927159031.pdb'),\n",
       " (19,\n",
       "  'MKKPRTSFTNEQLARLEREFNRQRYLTAQRRAQLSSELNLTEAQVKIWFQNKRAKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4835296252289497812.pdb'),\n",
       " (20,\n",
       "  'KPRTAFSSEQLARLKKEFNENRYLSARRRQQLAASLNLTEAQIKIWFQNRRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6780569998637110782.pdb'),\n",
       " (21,\n",
       "  'MKRPRTSFTSEQLARLEKEFNRQRYLTARRRAQLSSELNLTEAQVKIWFQNKRAKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7305603416206835138.pdb'),\n",
       " (22,\n",
       "  'KPRTSFTSEQLARLEREFNEQRYLTAQRRQELAAELNLTETQIKIWFQNKRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-606181091783216322.pdb'),\n",
       " (23,\n",
       "  'MKKPRTSFTSEQIARLEKEFNENRYLSEQRRQQLSAELGLNEAQVKIWFQNKRYKT',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2988697716310081023.pdb'),\n",
       " (24,\n",
       "  'KPRTAFTNEQLARLEREFNEQRYLSARRRQELSAELNLTEAQIKIWFQNRRAKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8412797332565811692.pdb'),\n",
       " (25,\n",
       "  'MKKPRTSFTSEQLARLKREFNRQRYLSAQRRQQLAASLNLNETQIKIWFQNRRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/4851238066339786443.pdb'),\n",
       " (26,\n",
       "  'KPRTAFSSEQLARLEREFNRQRYLSARRRQQLASSLNLTEAQIKIWFQNKRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1070938690462160174.pdb'),\n",
       " (27,\n",
       "  'KPRTSFTNEQLARLEREFNEQRYLSARRRQQLSAELNLTETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1114073927323066196.pdb'),\n",
       " (28,\n",
       "  'KPRTAFTNEQLARLEREFNEQRYLSARRRQELAAELNLTEAQIKIWFQNKRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1559454905206351569.pdb'),\n",
       " (29,\n",
       "  'KPRTSFTNEQLARLEREFNEQRYLSARRRQELAAELNLTETQIKIWFQNKRYKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5908434103173073659.pdb'),\n",
       " (30,\n",
       "  'RPRTSFTSEQLARLEKEFNENRYLSARRRAQLSSELNLTETQIKIWFQNRRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5926213234545361994.pdb'),\n",
       " (31,\n",
       "  'KPRTSFSSEQLARLEREFNENRYLTARRRQELAASLNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/1805987435303003842.pdb'),\n",
       " (32,\n",
       "  'KPRTAFSSEQLARLKKEFNENRYLSARRRQQLAASLNLNETQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4954729405650404590.pdb'),\n",
       " (33,\n",
       "  'KPRTSFSSEQLARLKKEFNENRYLSARRRQQLAASLNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7524046860829988900.pdb'),\n",
       " (34,\n",
       "  'KPRTSFSSEQLARLEREFNRQRYLTARRRQELASSLNLTETQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6480133699357489018.pdb'),\n",
       " (35,\n",
       "  'KPRTAFSSEQLARLKREFNRNRYLSARRRQQLAASLNLTETQIKIWFQNRRYKT',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6026618840856319822.pdb'),\n",
       " (36,\n",
       "  'KPRTAFTNEQLARLEKEFNENRYLSARRRQQLSAELNLTETQIKIWFQNKRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/304619159461822205.pdb'),\n",
       " (37,\n",
       "  'KPRTAFTNEQLARLEKEFNEQRYLTAQRRQQLSAELNLNEAQIKIWFQNRRAKI',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7742646998289812311.pdb'),\n",
       " (38,\n",
       "  'KPRTSFSSEQLARLKREFNRNRYLSARRRQQLSASLNLNETQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-740262411460968530.pdb'),\n",
       " (39,\n",
       "  'MKKPRTSFSSEQLARLEREFNRQRYLTARRRQELASSLNLTETQIKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8100890809300184631.pdb'),\n",
       " (40,\n",
       "  'KPRTAFSSEQLARLKKEFNRQRYLSARRRQQLAASLNLTETQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4550980616417252897.pdb'),\n",
       " (41,\n",
       "  'KPRTSFSSEQIARLEREFNENRYLTARRRQELAASLNLNEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-29632981891748370.pdb'),\n",
       " (42,\n",
       "  'KPRTSFSSEQLARLEREFNRQRYLTARRRQELASSLNLTETQIKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5978213321369291198.pdb'),\n",
       " (43,\n",
       "  'KPRTAFSNEQLARLEKEFNENRYLSARRRQELAASLNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6508733586595049341.pdb'),\n",
       " (44,\n",
       "  'KPRTAFSSEQLARLEREFNRNRYLSARRRQQLAASLNLTETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-8612689115092436570.pdb'),\n",
       " (45,\n",
       "  'KPRTSFSSEQLARLKREFNRQRYLTARRRQQLASSLNLTETQIKIWFQNRRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-9124443349131670794.pdb'),\n",
       " (46,\n",
       "  'RPRTAFTNEQLARLEKEFNENRYLSARRRQQLSAELNLTETQIKIWFQNKRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3653699920656111016.pdb'),\n",
       " (47,\n",
       "  'KPRTSFSSEQLARLEREFNENRYLTARRRQELAAELNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2894244318917440863.pdb'),\n",
       " (48,\n",
       "  'KPRTAFTNEQLARLEKEFNENRYLSARRRQQLSAELNLTETQIKIWFQNKRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/304619159461822205.pdb'),\n",
       " (49,\n",
       "  'KPRTSFTSEQLARLEKEFNRNRYLTARRRQQLASSLNLTETQIKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/7533046716735369174.pdb'),\n",
       " (50,\n",
       "  'KPRTSFSNEQLARLKREFNRNRYLSARRRQQLAASLNLNETQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1192725946674690642.pdb'),\n",
       " (51,\n",
       "  'KPRTAFSSEQLARLEKEFNRNRYLTARRRQQLASSLNLTETQIKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4193604207240227687.pdb'),\n",
       " (52,\n",
       "  'RPRTSFSSEQLARLEREFNRNRYLSARRRQQLAASLNLNETQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4220604839223820442.pdb'),\n",
       " (53,\n",
       "  'KPRTAFSSEQLARLEREFNENRYLSARRRQELASSLNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-8328985860777735211.pdb'),\n",
       " (54,\n",
       "  'RPRTSFSSEQIARLEKEFNENRYLSARRRQELAAELNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-5276196520896089480.pdb'),\n",
       " (55,\n",
       "  'KPRTSFTSEQLARLEKEFNRNRYLTARRRQQLASSLNLNETQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5597804988003154781.pdb'),\n",
       " (56,\n",
       "  'KPRTAFSSEQLARLKKEFNRNRYLSARRRQQLAASLNLTETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2752861838611097958.pdb'),\n",
       " (57,\n",
       "  'KPRTAFSSEQLARLEKEFNRNRYLTARRRQQLASSLNLTEAQIKIWFQNKRAKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1609998974780391243.pdb'),\n",
       " (58,\n",
       "  'RPRTSFTSEQLARLEKEFNENRYLSARRRAQLSSSLNLTETQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3625197254694812423.pdb'),\n",
       " (59,\n",
       "  'KPRTSFSNEQIARLEREFNENRYLTAQRRQELAASLNLNEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2659900203422731670.pdb'),\n",
       " (60,\n",
       "  'KPRTAFSSEQLARLEREFNRNRYLSARRRQQLAASLNLTEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4082372247685935402.pdb'),\n",
       " (61,\n",
       "  'KPRTSFSSEQLARLEREFNENRYLSARRRQQLAASLNLTETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6193626290068998265.pdb'),\n",
       " (62,\n",
       "  'KPRTSFSSEQIARLEKEFNENRYLSARRRQELAASLNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-9189675965504242013.pdb'),\n",
       " (63,\n",
       "  'KPRTAFSSEQLARLEKEFNENRYLTARRRQELAAELNLTEAQIKIWFQNRRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8308557137457216643.pdb'),\n",
       " (64,\n",
       "  'KPRTSFTSEQLARLEKEFNENRYLSARRRAQLASSLNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3589026986465486999.pdb'),\n",
       " (65,\n",
       "  'KPRTAFSNEQLARLEKEFNENRYLSARRRQELAASLNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6508733586595049341.pdb'),\n",
       " (66,\n",
       "  'KPRTSFSSEQLARLEREFNRNRYLSARRRQQLAAELNLTETQVKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7601903680001017983.pdb'),\n",
       " (67,\n",
       "  'KPRTSFSSEQLARLEREFNRNRYLSARRRQQLAAELNLNETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6290212184287624824.pdb'),\n",
       " (68,\n",
       "  'KPRTSFSSEQLARLEREFNENRYLSERRRQQLAASLNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5997900312004824970.pdb'),\n",
       " (69,\n",
       "  'KPRTSFSNEQLARLKREFNENRYLSARRRQQLAASLNLNETQVKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-7031242467403634670.pdb'),\n",
       " (70,\n",
       "  'KPRTSFSSEQLARLEREFNRNRYLTARRRQELAAELNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2770315071241457610.pdb'),\n",
       " (71,\n",
       "  'KPRTSFSSEQLARLEREFNENRYLSARRRQQLAAELNLTETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5807432384334425262.pdb'),\n",
       " (72,\n",
       "  'KPRTSFSSEQLARLEREFNRNRYLSARRRQQLAASLNLTEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8061840995252715136.pdb'),\n",
       " (73,\n",
       "  'KPRTAFSSEQLARLEREFNRNRYLSARRRQQLAAELNLTEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2848433030294144672.pdb'),\n",
       " (74,\n",
       "  'KPRTSFSSEQIARLEKEFNENRYLSARRRQELAASLNLTETQVKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1075470608787742272.pdb'),\n",
       " (75,\n",
       "  'KPRTSFSSEQLARLEREFNENRYLSARRRQQLAASLNLTETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6193626290068998265.pdb'),\n",
       " (76,\n",
       "  'KPRTAFSSEQLARLEKEFNRNRYLSARRRQELAASLNLTEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-2793202724647554855.pdb'),\n",
       " (77,\n",
       "  'KPRTSFSSEQLARLEKEFNENRYLSERRRQELAASLNLTEAQIKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8942573535722633589.pdb'),\n",
       " (78,\n",
       "  'KPRTSFTSEQLARLEKEFNENRYLSARRRQQLASSLNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-8027899906572423913.pdb'),\n",
       " (79,\n",
       "  'KPRTSFSSEQLARLEREFNENRYLSARRRQELAASLNLTEAQVKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-9204037960066299633.pdb'),\n",
       " (80,\n",
       "  'KPRTSFSSEQLARLEKEFNRNRYLSERRRQQLAASLNLTEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3038036755247622606.pdb'),\n",
       " (81,\n",
       "  'KPRTSFSSEQIARLEKEFNENRYLSARRRQQLAASLNLTEAQIKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-1987144150101405640.pdb'),\n",
       " (82,\n",
       "  'KPRTAFSSEQLARLEKEFNRNRYLSERRRQELAASLNLTEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4802459429825186356.pdb'),\n",
       " (83,\n",
       "  'KPRTSFSSEQLARLEKEFNENRYLSARRRQQLAASLNLTEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/539077892644054150.pdb'),\n",
       " (84,\n",
       "  'KPRTAFSSEQLARLEREFNRNRYLSARRRQQLAASLNLNETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-6214545272213767512.pdb'),\n",
       " (85,\n",
       "  'KPRTAFSSEQIARLEREFNRNRYLSARRRQELAASLNLNEAQIKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-4511594999902598389.pdb'),\n",
       " (86,\n",
       "  'KPRTSFSSEQLARLEREFNRNRYLSARRRQQLAASLNLTEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8061840995252715136.pdb'),\n",
       " (87,\n",
       "  'KPRTSFSSEQLARLEREFNRNRYLSARRRQQLAAELNLNETQVKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/4259389542862946544.pdb'),\n",
       " (88,\n",
       "  'KPRTSFSSEQIARLEREFNRNRYLSERRRQQLSAELNLNEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-331100060556709620.pdb'),\n",
       " (89,\n",
       "  'KPRTAFSSEQLARLEREFNRNRYLSARRRQQLAASLNLTEAQVKIWFQNKRYKTKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3805470843243447233.pdb'),\n",
       " (90,\n",
       "  'KPRTSFSSEQLARLEKEFNRNRYLSERRRQELAASLNLTEAQIKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/5683158093040863460.pdb'),\n",
       " (91,\n",
       "  'KPRTAFTSEQLARLEKEFNRNRYLSERRRQQLAASLNLTETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/8465266648052980399.pdb'),\n",
       " (92,\n",
       "  'KPRTSFSSEQLARLEKEFNRNRYLSARRRAQLAAELNLNEAQIKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-9184860667643465329.pdb'),\n",
       " (93,\n",
       "  'KPRTSFSSEQLARLEKEFNRNRYLSERRRQQLAASLNLTEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3038036755247622606.pdb'),\n",
       " (94,\n",
       "  'KPRTAFSSEQLARLEKEFNRNRYLSERRRQELAASLNLTEAQIKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3216684077076472947.pdb'),\n",
       " (95,\n",
       "  'KPRTSFSSEQLARLEKEFNENRYLSERRRQELAASLNLTEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6686977071321137246.pdb'),\n",
       " (96,\n",
       "  'KPRTSFSSEQLARLEREFNRNRYLSARRRQELAASLNLTETQVKIWFQNRRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/-3826548347567650677.pdb'),\n",
       " (97,\n",
       "  'KPRTAFSSEQLARLEREFNRNRYLSARRRQQLAAELNLTETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/2427785058200115687.pdb'),\n",
       " (98,\n",
       "  'KPRTSFSSEQLARLEREFNRNRYLSERRRQQLAASLNLNEAQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/3982298190510282372.pdb'),\n",
       " (99,\n",
       "  'KPRTSFSSEQLARLEREFNRNRYLSARRRQQLAASLNLTETQVKIWFQNKRYKIKR',\n",
       "  '/mmfs1/gscratch/cheme/usr/evankomp/repos/nomelt/analysis/tmp/mutation_optimizer_run/6949535832702939191.pdb')]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "optimizer.structure_seq_trajectory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "540715a1-6386-45cf-9adb-a5f1d2bff3d1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:filelock:Attempting to acquire lock 22373022625744 on /mmfs1/home/evankomp/.triton/autotune/Fp16Matmul_2d_kernel.pickle.lock\n",
      "DEBUG:filelock:Lock 22373022625744 acquired on /mmfs1/home/evankomp/.triton/autotune/Fp16Matmul_2d_kernel.pickle.lock\n",
      "DEBUG:filelock:Attempting to release lock 22373022625744 on /mmfs1/home/evankomp/.triton/autotune/Fp16Matmul_2d_kernel.pickle.lock\n",
      "DEBUG:filelock:Lock 22373022625744 released on /mmfs1/home/evankomp/.triton/autotune/Fp16Matmul_2d_kernel.pickle.lock\n",
      "DEBUG:filelock:Attempting to acquire lock 22367615600432 on /mmfs1/home/evankomp/.triton/autotune/Fp16Matmul_4d_kernel.pickle.lock\n",
      "DEBUG:filelock:Lock 22367615600432 acquired on /mmfs1/home/evankomp/.triton/autotune/Fp16Matmul_4d_kernel.pickle.lock\n",
      "DEBUG:filelock:Attempting to release lock 22367615600432 on /mmfs1/home/evankomp/.triton/autotune/Fp16Matmul_4d_kernel.pickle.lock\n",
      "DEBUG:filelock:Lock 22367615600432 released on /mmfs1/home/evankomp/.triton/autotune/Fp16Matmul_4d_kernel.pickle.lock\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "with open('./tmp/optimizer_traj.json', 'w') as f:\n",
    "    json.dump(optimizer.structure_seq_trajectory, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3cc9889b-c916-4b80-a770-f73891c72b37",
   "metadata": {},
   "outputs": [],
   "source": [
    "optimizer.study.trials_dataframe().to_csv('./tmp/optimizer_trials.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66fd6139-083c-497c-8954-8ccfe4db640d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
